<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Benchmark</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">TOC</a>
</li>
<li>
  <a href="introduction.html">1. Introduction</a>
</li>
<li>
  <a href="tutorial.html">2.1 Tutorial</a>
</li>
<li>
  <a href="benchmark.html">2.2 Benchmark</a>
</li>
<li>
  <a href="frameworks.html">2.3 Frameworks</a>
</li>
<li>
  <a href="model_api.html">2.4 Alternatives</a>
</li>
<li>
  <a href="conclusions.html">3. Conclusions</a>
</li>
<li>
  <a href="appendix.html">4. Appendix</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/schae211/report">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Benchmark</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211125code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211125)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211125code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211125)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Saez/report
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttreef032329d5e4c60d6a7de2866da60713b5a76cadetargetblankf032329a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/schae211/report/tree/f032329d5e4c60d6a7de2866da60713b5a76cade" target="_blank">f032329</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttreef032329d5e4c60d6a7de2866da60713b5a76cadetargetblankf032329a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/schae211/report/tree/f032329d5e4c60d6a7de2866da60713b5a76cade" target="_blank">f032329</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/2021-12-06-15-42_all_results.RDS
    Ignored:    data/2021-12-08-14-10_all_results.RDS
    Ignored:    data/2021-12-09-09-21_all_results.RDS
    Ignored:    data/2021-12-09-17-26_all_results.RDS
    Ignored:    data/2021-12-13-18-43_all_results.RDS
    Ignored:    data/2021-12-14-10-57_all_views.RDS
    Ignored:    data/2021-12-14-17-59_all_views.RDS
    Ignored:    data/2021-12-14-23-01_all_results.RDS
    Ignored:    results/

Untracked files:
    Untracked:  LICENSE.md
    Untracked:  analysis/appendix.Rmd
    Untracked:  analysis/conclusions.Rmd
    Untracked:  analysis/introduction.Rmd
    Untracked:  analysis/references.bib
    Untracked:  analysis/science.csl
    Untracked:  cluster_scripts/bash_scripts/four_i.sh
    Untracked:  cluster_scripts/bash_scripts/two_args_caller.sh
    Untracked:  cluster_scripts/benchmark/four_i.R
    Untracked:  cluster_scripts/benchmark/synthetic.R
    Untracked:  cluster_scripts/preprocessing/synthetic_prep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/benchmark.Rmd</code>) and HTML (<code>docs/benchmark.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f032329d5e4c60d6a7de2866da60713b5a76cade/analysis/benchmark.Rmd" target="_blank">f032329</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/benchmark.html" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f14d98f689925e2d93cad95d5c043dd103238cec/analysis/benchmark.Rmd" target="_blank">f14d98f</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/421d5d44f065639498503f68193015533d86883b/docs/benchmark.html" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/7e810a54c78c0c06c2065e5dac02180c235bca9b/analysis/benchmark.Rmd" target="_blank">7e810a5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
workflowr::wflow_publish(“analysis/benchmark.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/benchmark.html" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/d83b8be0d9604a14b58ba504f3855ac91d668bc2/analysis/benchmark.Rmd" target="_blank">d83b8be</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
workflowr::wflow_publish(“analysis/benchmark.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/487e9ef41f3e9db110529e2e87697965ec804547/docs/benchmark.html" target="_blank">487e9ef</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/8fcde1bffefbbabbea7647ae9778e92062513766/analysis/benchmark.Rmd" target="_blank">8fcde1b</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
<td>
wflow_publish(“analysis/benchmark.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/067350bf869d53ad7ccb91c13d49c77e2d78860d/docs/benchmark.html" target="_blank">067350b</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/b68d23d819a8167d19093d76853ba64c512ca32d/docs/benchmark.html" target="_blank">b68d23d</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/565f4e02511a796a9de49f418982b5a8cb298745/analysis/benchmark.Rmd" target="_blank">565f4e0</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/a450bedf5080c68a7df841975f8f12351c37827f/docs/benchmark.html" target="_blank">a450bed</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/e197adb7350543bca1877b7f0422f9948151f382/analysis/benchmark.Rmd" target="_blank">e197adb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
wflow_publish(files = “analysis/benchmark.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Setup.</p>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE)
knitr::opts_knit$set(root.dir = &quot;~/Saez/report&quot;)</code></pre>
<pre class="r"><code>suppressPackageStartupMessages(library(mistyR))
suppressPackageStartupMessages(library(tidyverse))</code></pre>
<div id="introduction-and-data" class="section level1">
<h1>Introduction and Data</h1>
<p>What data is the benchmarking based on?</p>
<pre class="r"><code>all.views &lt;- readRDS(&quot;data/2021-12-14-17-59_all_views.RDS&quot;)
all.results &lt;- readRDS(&quot;data/2021-12-14-23-01_all_results.RDS&quot;)
all.names &lt;- names(all.results)
experiments &lt;- str_extract(all.names, &quot;[^/]+(?=/)&quot;) %&gt;% unique
experiments
[1] &quot;4iANCA&quot;           &quot;merfish_bc&quot;       &quot;merfish_liver&quot;    &quot;merfish_preoptic&quot;
[5] &quot;mibi_tnbc&quot;        &quot;synthetic&quot;       </code></pre>
<p>Where do the data come from?</p>
<div id="a-synthetic" class="section level2">
<h2>a) <code>synthetic</code></h2>
<p>    Simulated data that are included in the MISTy package. As explained in the <a href="https://saezlab.github.io/mistyR/articles/mistyR.html">Get Started Vignette</a>,this dataset is based on a two-dimensional cellular automata model which models the production, diffusion, degradation and interactions of 11 molecular species in 4 different cell types.</p>
<ul>
<li>  Number of Samples, Markers, and spatial units per sample</li>
</ul>
<pre class="r"><code># Number of Samples
c(length(all.views$`/synthetic/l6`),
  length(all.views$`/synthetic/l12`),
  length(all.views$`/synthetic/l24`)
  )
[1] 10 10 10

# Number of Markers
ncol(all.views$`/synthetic/l6`[[1]]$intraview$data)
[1] 11

# Summary for number of cells per sample
summary(
  map_dbl(all.views$`/synthetic/l6`, 
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   4110    4120    4148    4152    4179    4205 </code></pre>
<p>    Features: Spatial Proteomics, Single Cell Resolution</p>
</div>
<div id="b-mibi_tnbc" class="section level2">
<h2>b) <code>mibi_tnbc</code></h2>
<ul>
<li><p>  Multiplexed Ion Beam Imaging (spatial proteomics) dataset from <a href="https://dx.doi.org/10.1016/j.cell.2018.08.039">A Structured Tumor-Immune Microenvironment in Triple Negative Breast Cancer Revealed by Multiplexed Ion Beam Imaging</a> by Keren, L. et al. (Cell 174, 1373-1387.e19 (2018)) <span class="citation">(<em>1</em>)</span></p></li>
<li><p>  Data Availability: <a href="https://mibi-share.ionpath.com/">https://mibi-share.ionpath.com/</a></p></li>
<li><p>  Features: Spatial Proteomics, Single Cell Resolution</p></li>
<li><p>  Number of Samples, Markers, and spatial units per sample</p></li>
</ul>
<pre class="r"><code># Number of Samples
length(all.views$`/mibi_tnbc/standard_views`)
[1] 40

# Number of Markers
ncol(all.views$`/mibi_tnbc/standard_views`[[1]]$intraview$data)
[1] 28

# Summary for number of cells per sample
summary(
  map_dbl(all.views$`/mibi_tnbc/standard_views`, 
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2046    3876    4961    4942    6080    8212 </code></pre>
</div>
<div id="c-merfish_preoptic" class="section level2">
<h2>c) <code>merfish_preoptic</code></h2>
<ul>
<li><p>  MERFISH data from the murine hypothalamic preoptic region from <a href="https://dx.doi.org/10.1126/science.aau5324">Molecular, spatial, and functional single-cell profiling of the hypothalamic preoptic region</a> by Moffitt, R. et al. (Science 362, 6416 (2018))</p></li>
<li><p>  Features: Spatial Transcriptomics, Single Cell Resolution</p></li>
<li><p>  Data Availability: <a href="http://www.spatialomics.org/SpatialDB/merfish_30385464_browse.php">Spatial DB</a></p></li>
<li><p>  Number of Samples, Markers, and spatial units per sample</p></li>
</ul>
<pre class="r"><code># Number of Samples
c(length(all.views$`/merfish_preoptic/hvg_views`),
  length(all.views$`/merfish_preoptic/standard_views`)
)
[1] 36 36

# Number of Markers
ncol(all.views$`/merfish_preoptic/hvg_views`[[1]]$intraview$data)
[1] 108

# Summary for number of cells per sample
summary(
  map_dbl(all.views$`/merfish_preoptic/hvg_views`, 
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  10291   21938   22642   28551   32065   73655 </code></pre>
</div>
<div id="d-merfish_liver" class="section level2">
<h2>d) <code>merfish_liver</code></h2>
<ul>
<li><p>  MERFISH data from the murine fetal liver from <a href="https://dx.doi.org/10.1038/s41421-021-00266-1">Spatial transcriptome profiling by MERFISH reveals fetal liver hematopoietic stem cell niche architecture</a> by Yanfang, L. et al. (Cell Discovery 7, 47 (2021))</p></li>
<li><p>  Features: Spatial Transcriptomics, Single Cell Resolution</p></li>
<li><p>  Data Availability: <a href="http://www.spatialomics.org/SpatialDB/merfish_30385464_browse.php">Spatial DB</a></p></li>
<li><p>  Number of Samples, Markers, and spatial units per sample</p></li>
</ul>
<pre class="r"><code># Number of Samples
c(length(all.views$`/merfish_liver/standard280_views`)
)
[1] 164

# Number of Markers
ncol(all.views$`/merfish_liver/standard280_views`[[1]]$intraview$data)
[1] 140

# Summary for number of cells per sample
summary(
  map_dbl(all.views$`/merfish_liver/standard280_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  280.0   296.8   315.0   315.4   334.0   366.0 </code></pre>
</div>
<div id="e-merfish_bc" class="section level2">
<h2>e) <code>merfish_bc</code></h2>
<ul>
<li><p>  MERFISH data from metastatic breast cancer (unpublished)</p></li>
<li><p>  Features: Spatial Transcriptomics, Single Cell Resolution</p></li>
<li><p>  Data Availability: <a href="http://www.spatialomics.org/SpatialDB/merfish_30385464_browse.php">Spatial DB</a></p></li>
<li><p>  Number of Samples, Markers, and spatial units per sample</p></li>
</ul>
<pre class="r"><code># Number of Samples
c(length(all.views$`/merfish_bc/hvg_views`),
  length(all.views$`/merfish_bc/standard_views`)
)
[1] 9 9

# Number of Markers
ncol(all.views$`/merfish_bc/hvg_views`[[1]]$intraview$data)
[1] 86

# Summary for number of cells per sample
summary(
  map_dbl(all.views$`/merfish_bc/hvg_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2704    3981   11014   11787   13822   31197 </code></pre>
</div>
<div id="f-4ianca" class="section level2">
<h2>f) <code>4iANCA</code></h2>
<ul>
<li><p>  4i (iterative indirect immunofluorescence imaging) data from glomeruli</p></li>
<li><p>  Features: Spatial Proteomics, Single Cell Resolution</p></li>
<li><p>  Data Availability: <a href="http://www.spatialomics.org/SpatialDB/merfish_30385464_browse.php">Spatial DB</a></p></li>
<li><p>  Number of Samples, Markers, and spatial units per sample</p></li>
</ul>
<pre class="r"><code># Number of Samples
c(length(all.views$`/4iANCA/ratio_153_views`),
  length(all.views$`/4iANCA/ratio_76_views`),
  length(all.views$`/4iANCA/ratio_38_views`),
  length(all.views$`/4iANCA/ratio_19_views`)
)
[1] 36 36 36 36

# Number of Markers
ncol(all.views$`/4iANCA/ratio_153_views`[[1]]$intraview$data)
[1] 27

# Summary for number of cells per sample
# Largest Binning
summary(
  map_dbl(all.views$`/4iANCA/ratio_153_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    169     169     169     169     169     169 

#  Binning
summary(
  map_dbl(all.views$`/4iANCA/ratio_76_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    676     676     676     676     676     676 

# Largest Binning
summary(
  map_dbl(all.views$`/4iANCA/ratio_38_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2704    2704    2704    2704    2704    2704 

# Largest Binning
summary(
  map_dbl(all.views$`/4iANCA/ratio_19_views`,
          ~ nrow(.x$intraview$data))
)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  10609   10609   10609   10609   10609   10609 </code></pre>
<ul>
<li>  Number of cells per sample: No segmentation was performed, therefore the pixels were simply binned with 3 different schemes. The pixels have a side length of <code>0.13</code> microns, and an average eukaryotic cell is between <code>10</code> and <code>100</code> microns in diameter. Therefore I choose the largest binning to be <code>floor(20/0.13) = 153</code>, meaning the side length of one spatial unit (bin) will correspond to 153 pixels. The smaller bins are then sequentially half of the size (76, 38, 19). The corresponding parameter for the computation of the paraview was adjusted accordingly, meaning multiplied sequentially by two.</li>
</ul>
</div>
</div>
<div id="scripts" class="section level1">
<h1>Scripts</h1>
<p>The scripts that were used for these analysis can be found in this <a href="https://github.com/schae211/report/tree/master/cluster_scripts">GitHub repository</a>. Theses scripts were run on a cluster managed by Slurm.</p>
</div>
<div id="comparing-algorithms" class="section level1">
<h1>Comparing Algorithms</h1>
<p>Function to get clean data for a particular study.</p>
<pre class="r"><code>filter_runs &lt;- function(all.names, all.runs, str.study) {
  names.run &lt;- all.names[str_detect(all.names, str.study)]
  successful.run &lt;- map_lgl(all.results[names.run], ~ typeof(.x) == &quot;list&quot;)
  all.results[names.run][successful.run]
}</code></pre>
<p>Function to extract the improvements (performance measures).</p>
<pre class="r"><code>get_performance &lt;- function(filtered.run) {
  map2_dfr(filtered.run, names(filtered.run),
                                   function(misty.run, name) {
    misty.run$improvements %&gt;%
      mutate(algorithm = str_extract(name, &quot;(?&lt;=/)[^/]+$&quot;)) %&gt;%
      mutate(study = str_extract(sample, &quot;(?&lt;=OUTPUT/)[^/]+&quot;))
  })
}</code></pre>
<div id="merfish-breast-cancer-spatially-variable-genes" class="section level2">
<h2>Merfish Breast Cancer (spatially variable genes)</h2>
<p>Cleaning Data.</p>
<pre class="r"><code>merfish_bc_hvg &lt;- filter_runs(all.names, all.results, &quot;/merfish_bc/hvg/&quot;)</code></pre>
<p>Performance.</p>
<pre class="r"><code>performance_merfish_bc_hvg &lt;- get_performance(merfish_bc_hvg)

performance_merfish_bc_hvg %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = factor(algorithm,
             levels=c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;LBOOST&quot;, &quot;MARS&quot;, &quot;LM&quot;, &quot;BGMARS&quot;))) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-12-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-12-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/487e9ef41f3e9db110529e2e87697965ec804547/docs/figure/benchmark.Rmd/unnamed-chunk-12-1.png" target="_blank">487e9ef</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="merfish-fetal-liver-spatially-variable-genes" class="section level2">
<h2>Merfish Fetal Liver (spatially variable genes)</h2>
<pre class="r"><code>merfish_liver_std280 &lt;- filter_runs(all.names, all.results, &quot;/merfish_liver/std280/&quot;)</code></pre>
<pre class="r"><code>performance_merfish_liver_std280 &lt;- get_performance(merfish_liver_std280)

performance_merfish_liver_std280 %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = factor(algorithm,
             levels=c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;LBOOST&quot;, &quot;MARS&quot;, &quot;LM&quot;, &quot;BGMARS&quot;))) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-14-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-14-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/487e9ef41f3e9db110529e2e87697965ec804547/docs/figure/benchmark.Rmd/unnamed-chunk-14-1.png" target="_blank">487e9ef</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="merfish-preoptic-spatially-variable-genes" class="section level2">
<h2>Merfish Preoptic (spatially variable genes)</h2>
<pre class="r"><code>merfish_preoptic_hvg &lt;- filter_runs(all.names, all.results,
                                    &quot;/merfish_preoptic/hvg/&quot;)</code></pre>
<pre class="r"><code>performance_merfish_preoptic_hvg &lt;- get_performance(merfish_preoptic_hvg)

performance_merfish_preoptic_hvg %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = factor(algorithm, 
                            levels=c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;LBOOST&quot;, &quot;LM&quot;, &quot;MARS&quot;, &quot;MARS80&quot;, &quot;MARS60&quot;, &quot;MARS40&quot;))) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-16-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mibi-tnbc" class="section level2">
<h2>MIBI TNBC</h2>
<pre class="r"><code>mibi_tnbc_std &lt;- filter_runs(all.names, all.results,
                                    &quot;/mibi_tnbc/standard/&quot;)</code></pre>
<pre class="r"><code>performance_mibi_tnbc_std &lt;- get_performance(mibi_tnbc_std)

performance_mibi_tnbc_std %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = factor(algorithm,
             levels=c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;LBOOST&quot;, &quot;MARS&quot;, &quot;LM&quot;, &quot;BGMARS&quot;))) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-18-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We see some overfitting here with linear boosting, since it is worse than the normal linear model.</p>
</div>
<div id="i" class="section level2">
<h2>4I</h2>
<pre class="r"><code># tmp
fouri_anca &lt;- all.results[str_starts(all.names, 
                                     &quot;/4iANCA&quot;)]
filtered_fouri_anca &lt;- fouri_anca[str_ends(names(fouri_anca), 
                                           &quot;BGMARS|SVM|MLP&quot;, 
                                           negate = TRUE)]
saveRDS(filtered_fouri_anca, &quot;../4iANCA_results.RDS&quot;)
str_extract(names(filtered_fouri_anca), &quot;ratio_[0-9]+&quot;) %&gt;% unique
[1] &quot;ratio_12&quot;  &quot;ratio_153&quot; &quot;ratio_19&quot;  &quot;ratio_24&quot;  &quot;ratio_38&quot;  &quot;ratio_6&quot;  
[7] &quot;ratio_76&quot; </code></pre>
<pre class="r"><code>iiii_std &lt;- filter_runs(all.names, all.results,
                                    &quot;/4iANCA/ratio_24/&quot;)</code></pre>
<pre class="r"><code>performance_iiii_std &lt;- get_performance(iiii_std)

l = c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;MARS100&quot;, &quot;LM&quot;, &quot;LBOOST&quot;)

performance_iiii_std %&gt;%
  filter(measure == &quot;gain.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Gain.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-21-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-21-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-21-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-21-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>performance_iiii_std %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-22-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>iiii_153 &lt;- filter_runs(all.names, all.results,
                                    &quot;/4iANCA/ratio_153/&quot;)</code></pre>
<pre class="r"><code>performance_iiii_153 &lt;- get_performance(iiii_153)

l = c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;MARS100&quot;, &quot;LM&quot;, &quot;LBOOST&quot;)

performance_iiii_153 %&gt;%
  filter(measure == &quot;gain.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Gain.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-24-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-24-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-24-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>performance_iiii_153 %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-25-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-25-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>iiii_76 &lt;- filter_runs(all.names, all.results,
                                    &quot;/4iANCA/ratio_76/&quot;)</code></pre>
<pre class="r"><code>performance_iiii_76 &lt;- get_performance(iiii_76)

l = c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;MARS100&quot;, &quot;LM&quot;, &quot;LBOOST&quot;)

performance_iiii_76 %&gt;%
  filter(measure == &quot;gain.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Gain.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-27-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-27-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-27-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-27-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>performance_iiii_76 %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  filter(algorithm %in% l) %&gt;%
  mutate(algorithm = factor(algorithm,
         levels = l)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-28-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-28-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="hyperparameter-optimization" class="section level1">
<h1>Hyperparameter Optimization</h1>
<div id="rf" class="section level2">
<h2>RF</h2>
<div id="mibi-tnbc-1" class="section level3">
<h3>MIBI TNBC</h3>
<pre class="r"><code>mibi_tnbc_rf_hyper &lt;- filter_runs(all.names, all.results,
                                    &quot;/mibi_tnbc/RF_hyper/&quot;)</code></pre>
<pre class="r"><code>performance_mibi_tnbc_rf_hyper &lt;- get_performance(mibi_tnbc_rf_hyper)

performance_mibi_tnbc_rf_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=RF_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-30-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-30-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="mars" class="section level2">
<h2>MARS</h2>
<div id="mibi-tnbc-2" class="section level3">
<h3>MIBI TNBC</h3>
<pre class="r"><code>mibi_tnbc_mars_hyper &lt;- filter_runs(all.names, all.results,
                                    &quot;/mibi_tnbc/MARS_hyper/&quot;)</code></pre>
<pre class="r"><code>performance_mibi_tnbc_mars_hyper &lt;- get_performance(mibi_tnbc_mars_hyper)

performance_mibi_tnbc_mars_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=MARS_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = median)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Median Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/figure/benchmark.Rmd/unnamed-chunk-32-1.png" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>performance_mibi_tnbc_mars_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=MARS_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = mean)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Mean Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/421d5d44f065639498503f68193015533d86883b/docs/figure/benchmark.Rmd/unnamed-chunk-33-1.png" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/figure/benchmark.Rmd/unnamed-chunk-33-1.png" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="synthetic" class="section level3">
<h3>Synthetic</h3>
<pre class="r"><code>synthetic_mars_hyper &lt;- filter_runs(all.names, all.results,
                                    &quot;/synthetic/MARS_hyper/&quot;)</code></pre>
<pre class="r"><code>performance_synthetic_mars_hyper &lt;- get_performance(synthetic_mars_hyper)

performance_synthetic_mars_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=MARS_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = mean)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  #theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Mean Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="linear-boosting" class="section level2">
<h2>Linear Boosting</h2>
<div id="mibi-tnbc-3" class="section level3">
<h3>MIBI TNBC</h3>
<pre class="r"><code>mibi_tnbc_lboost_hyper &lt;- filter_runs(all.names, all.results,
                                    &quot;/mibi_tnbc/LBOOST_hyper/&quot;)</code></pre>
<pre class="r"><code>performance_mibi_tnbc_lboost_hyper &lt;- get_performance(mibi_tnbc_lboost_hyper)

performance_mibi_tnbc_lboost_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=LBOOST_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = mean)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Mean Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tree-boosting" class="section level2">
<h2>Tree Boosting</h2>
<div id="mibi-tnbc-4" class="section level3">
<h3>MIBI TNBC</h3>
<pre class="r"><code>mibi_tnbc_tboost_hyper &lt;- filter_runs(all.names, all.results,
                                    &quot;/mibi_tnbc/TBOOST_hyper/&quot;)</code></pre>
<pre class="r"><code>performance_mibi_tnbc_tboost_hyper &lt;- get_performance(mibi_tnbc_tboost_hyper)

performance_mibi_tnbc_tboost_hyper %&gt;%
  filter(measure == &quot;multi.R2&quot;) %&gt;%
  mutate(algorithm = str_extract(algorithm, &quot;(?&lt;=TBOOST_hyper_).+&quot;)) %&gt;%
  group_by(study, algorithm, target) %&gt;%
  summarise(mean = mean(value), median = median(value), 
            sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
  ggplot() +
  geom_tile(aes(x = algorithm, y = target, fill = mean)) +
  #facet_wrap(~ study, scales=&quot;free&quot;, nrow=1) +
  scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank()) +
  labs(x = &quot;Algorithm&quot;, fill = &quot;Mean Multi.R2&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="tmp" class="section level1">
<h1>tmp</h1>
<pre class="r"><code>all.results$`/merfish_preoptic/hvg/MARS`$improvements %&gt;%
  pull(target) %&gt;% unique
[1] &quot;Sema3c&quot; &quot;Trh&quot;    &quot;Sox6&quot;  

all.results$`/merfish_preoptic/hvg/RF`$improvements %&gt;%
  pull(target) %&gt;% unique
  [1] &quot;Adcyap1&quot;   &quot;Gpr165&quot;    &quot;Dgkk&quot;      &quot;Gda&quot;       &quot;Cckar&quot;     &quot;Calcr&quot;    
  [7] &quot;Sp9&quot;       &quot;Fst&quot;       &quot;Arhgap36&quot;  &quot;Tac1&quot;      &quot;Col25a1&quot;   &quot;Pcdh11x&quot;  
 [13] &quot;Tmem108&quot;   &quot;Mlc1&quot;      &quot;Lmod1&quot;     &quot;Slc17a6&quot;   &quot;Opalin&quot;    &quot;Npas1&quot;    
 [19] &quot;Pou3f2&quot;    &quot;Sema3c&quot;    &quot;Ebf3&quot;      &quot;Greb1&quot;     &quot;Cckbr&quot;     &quot;Gabra1&quot;   
 [25] &quot;Baiap2&quot;    &quot;Gem&quot;       &quot;Aldh1l1&quot;   &quot;Tac2&quot;      &quot;Cartpt&quot;    &quot;Cpne5&quot;    
 [31] &quot;Prlr&quot;      &quot;Synpr&quot;     &quot;Sema4d&quot;    &quot;Slc18a2&quot;   &quot;Ndrg1&quot;     &quot;Penk&quot;     
 [37] &quot;Trh&quot;       &quot;Oprd1&quot;     &quot;Lpar1&quot;     &quot;Npy1r&quot;     &quot;Gabrg1&quot;    &quot;Sln&quot;      
 [43] &quot;Htr2c&quot;     &quot;Amigo2&quot;    &quot;Bdnf&quot;      &quot;Rgs2&quot;      &quot;Ermn&quot;      &quot;Ccnd2&quot;    
 [49] &quot;Pgr&quot;       &quot;Serpinb1b&quot; &quot;Creb3l1&quot;   &quot;Syt2&quot;      &quot;Gjc3&quot;      &quot;Cbln1&quot;    
 [55] &quot;Tacr1&quot;     &quot;Trhr&quot;      &quot;Oprk1&quot;     &quot;Necab1&quot;    &quot;Npy2r&quot;     &quot;Man1a&quot;    
 [61] &quot;Gad1&quot;      &quot;Igf2r&quot;     &quot;Rnd3&quot;      &quot;Oprl1&quot;     &quot;Esr1&quot;      &quot;Sox4&quot;     
 [67] &quot;Syt4&quot;      &quot;Ano3&quot;      &quot;Glra3&quot;     &quot;Cspg5&quot;     &quot;Cd24a&quot;     &quot;Serpine1&quot; 
 [73] &quot;Cbln2&quot;     &quot;Nnat&quot;      &quot;Bmp7&quot;      &quot;Ttyh2&quot;     &quot;Plin3&quot;     &quot;Mbp&quot;      
 [79] &quot;Tacr3&quot;     &quot;Ntng1&quot;     &quot;Pak3&quot;      &quot;Gal&quot;       &quot;Irs4&quot;      &quot;Scg2&quot;     
 [85] &quot;Sytl4&quot;     &quot;Cyp19a1&quot;   &quot;Gnrh1&quot;     &quot;Etv1&quot;      &quot;Aqp4&quot;      &quot;Cck&quot;      
 [91] &quot;Sox6&quot;      &quot;Nos1&quot;      &quot;Brs3&quot;      &quot;Ucn3&quot;      &quot;Chat&quot;      &quot;Pnoc&quot;     
 [97] &quot;Sgk1&quot;      &quot;Nts&quot;       &quot;Mc4r&quot;      &quot;Tiparp&quot;    &quot;Isl1&quot;      &quot;Scgn&quot;     
[103] &quot;Ar&quot;        &quot;Fezf1&quot;     &quot;Coch&quot;      &quot;Sox8&quot;      &quot;Vgf&quot;       &quot;Onecut2&quot;  </code></pre>
<pre class="r"><code>x &lt;- sample(c(rep(&quot;hello&quot;, 10), rep(&quot;world&quot;, 10), rep(&quot;idea&quot;, 2)))
x
 [1] &quot;hello&quot; &quot;world&quot; &quot;hello&quot; &quot;hello&quot; &quot;hello&quot; &quot;world&quot; &quot;world&quot; &quot;world&quot; &quot;world&quot;
[10] &quot;hello&quot; &quot;idea&quot;  &quot;world&quot; &quot;world&quot; &quot;hello&quot; &quot;hello&quot; &quot;world&quot; &quot;world&quot; &quot;hello&quot;
[19] &quot;hello&quot; &quot;hello&quot; &quot;world&quot; &quot;idea&quot; 

case_when(x == &quot;hello&quot; ~ &quot;bar&quot;,
          x == &quot;world&quot; ~ &quot;foo&quot;,
          TRUE ~ &quot;else&quot;)
 [1] &quot;bar&quot;  &quot;foo&quot;  &quot;bar&quot;  &quot;bar&quot;  &quot;bar&quot;  &quot;foo&quot;  &quot;foo&quot;  &quot;foo&quot;  &quot;foo&quot;  &quot;bar&quot; 
[11] &quot;else&quot; &quot;foo&quot;  &quot;foo&quot;  &quot;bar&quot;  &quot;bar&quot;  &quot;foo&quot;  &quot;foo&quot;  &quot;bar&quot;  &quot;bar&quot;  &quot;bar&quot; 
[21] &quot;foo&quot;  &quot;else&quot;</code></pre>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-10.1016/j.cell.2018.08.039" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">L. Keren, M. Bosse, D. Marquez, R. Angoshtari, S. Jain, S. Varma, S.-R. Yang, A. Kurian, D. V. Valen, R. West, S. C. Bendall, M. Angelo, <span class="nocase">A Structured Tumor-Immune Microenvironment in Triple Negative Breast Cancer Revealed by Multiplexed Ion Beam Imaging</span>. <em>Cell</em>. <strong>174</strong>, 1373–1387.e19 (2018).</div>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4    
 [5] readr_2.0.2     tidyr_1.1.4     tibble_3.1.5    ggplot2_3.3.5  
 [9] tidyverse_1.3.1 mistyR_1.3.1    workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        lubridate_1.8.0   assertthat_0.2.1  rprojroot_2.0.2  
 [5] digest_0.6.28     utf8_1.2.2        R6_2.5.1          cellranger_1.1.0 
 [9] backports_1.3.0   reprex_2.0.1      evaluate_0.14     httr_1.4.2       
[13] highr_0.9         pillar_1.6.3      rlang_0.4.12      readxl_1.3.1     
[17] rstudioapi_0.13   whisker_0.4       jquerylib_0.1.4   rmarkdown_2.11   
[21] labeling_0.4.2    munsell_0.5.0     broom_0.7.9       compiler_4.1.2   
[25] httpuv_1.6.3      modelr_0.1.8      xfun_0.26         pkgconfig_2.0.3  
[29] htmltools_0.5.2   tidyselect_1.1.1  viridisLite_0.4.0 fansi_0.5.0      
[33] crayon_1.4.1      tzdb_0.1.2        dbplyr_2.1.1      withr_2.4.2      
[37] later_1.3.0       grid_4.1.2        jsonlite_1.7.2    gtable_0.3.0     
[41] lifecycle_1.0.1   DBI_1.1.1         git2r_0.28.0      magrittr_2.0.1   
[45] scales_1.1.1      cli_3.0.1         stringi_1.7.5     farver_2.1.0     
[49] fs_1.5.0          promises_1.2.0.1  xml2_1.3.2        bslib_0.3.1      
[53] ellipsis_0.3.2    generics_0.1.0    vctrs_0.3.8       tools_4.1.2      
[57] glue_1.4.2        hms_1.1.1         fastmap_1.1.0     yaml_2.2.1       
[61] colorspace_2.0-2  rvest_1.0.1       knitr_1.36        haven_2.4.3      
[65] sass_0.4.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
