<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Benchmark</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">TOC</a>
</li>
<li>
  <a href="introduction.html">1. Introduction</a>
</li>
<li>
  <a href="tutorial.html">2. Methods</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    3. Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="benchmark.html">3.1 Benchmark</a>
    </li>
    <li>
      <a href="frameworks.html">3.2 Frameworks</a>
    </li>
    <li>
      <a href="model_api.html">3.3 Alternatives</a>
    </li>
    <li>
      <a href="case_study_A.html">3.4 Case Study A</a>
    </li>
    <li>
      <a href="case_study_B.html">3.5 Case Study B</a>
    </li>
  </ul>
</li>
<li>
  <a href="conclusions.html">4. Conclusions</a>
</li>
<li>
  <a href="appendix.html">5. Appendix</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/schae211/report">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Benchmark</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211125code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211125)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211125code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211125)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttree498e920eef1cecd04ccfe98cff78292243e755dftargetblank498e920a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/schae211/report/tree/498e920eef1cecd04ccfe98cff78292243e755df" target="_blank">498e920</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttree498e920eef1cecd04ccfe98cff78292243e755dftargetblank498e920a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/schae211/report/tree/498e920eef1cecd04ccfe98cff78292243e755df" target="_blank">498e920</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/
    Ignored:    data/4iANCA/
    Ignored:    data/imc/
    Ignored:    data/merfish_bc/
    Ignored:    data/merfish_liver/
    Ignored:    data/merfish_preoptic/
    Ignored:    data/mibi_tnbc/
    Ignored:    data/seqfishp_brain/
    Ignored:    output/2021-12-17-07-58_all_results.RDS
    Ignored:    output/2021-12-17-07-58_all_results.log
    Ignored:    output/2021-12-17-07-59_all_views.RDS
    Ignored:    output/2021-12-20-16-20_all_results.RDS
    Ignored:    output/2021-12-20-16-20_all_results.log
    Ignored:    output/2021-12-21-12-22_all_results.RDS
    Ignored:    output/2021-12-21-12-22_all_results.log
    Ignored:    output/2021-12-23-09-57_all_results.RDS
    Ignored:    output/2021-12-23-09-57_all_results.log

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/benchmark.Rmd</code>) and HTML (<code>docs/benchmark.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/498e920eef1cecd04ccfe98cff78292243e755df/analysis/benchmark.Rmd" target="_blank">498e920</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
<td>
changing to relative paths
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/64a9eda59c4bb96fda82c0cc7316b34a89a794b7/analysis/benchmark.Rmd" target="_blank">64a9eda</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
<td>
changing folder structure and .gitignore
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/9229052d2535073c992e78722393aa181c209ec7/docs/benchmark.html" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f9a0c99135bb73dc77bf3711c5a273e86695ae0c/analysis/benchmark.Rmd" target="_blank">f9a0c99</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
<td>
wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/benchmark.html" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f69aed2f021cef1d9f461af7f50319e59debdb6a/analysis/benchmark.Rmd" target="_blank">f69aed2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
<td>
wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/bbf04bbb5e42a9c22992a5299c8cf0c5fc22b99f/docs/benchmark.html" target="_blank">bbf04bb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/47ffb40edbf29bc5caa3a20daca1c3d19f82fe71/analysis/benchmark.Rmd" target="_blank">47ffb40</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
<td>
wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/781ba638f2acfdc32de627718c184805cc4c9048/docs/benchmark.html" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/ef0bde60c5de5a12bdb01338e17b12c9e1112a5e/analysis/benchmark.Rmd" target="_blank">ef0bde6</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/benchmark.html" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/c5d14b74afd9918f2317205e6772357ec81d4544/analysis/benchmark.Rmd" target="_blank">c5d14b7</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
<td>
workflowr::wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/e9c007647ea473f2118b7e74517932deb1b93556/docs/benchmark.html" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/abb88bb220a0839e3f77381ec9ceabaff514905f/analysis/benchmark.Rmd" target="_blank">abb88bb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/benchmark.html" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/45f967b42c1b18f6b88eca4b0d05149da5254556/analysis/benchmark.Rmd" target="_blank">45f967b</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/1710d804182e867e8d3dc0d236fbf2f976201c9e/docs/benchmark.html" target="_blank">1710d80</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/f72d59c3b66e424cd5f45999439ddcddcf91bbfc/docs/benchmark.html" target="_blank">f72d59c</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f032329d5e4c60d6a7de2866da60713b5a76cade/analysis/benchmark.Rmd" target="_blank">f032329</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/c428797136ab78c4c3b44f20bffb8556bf7e1d72/docs/benchmark.html" target="_blank">c428797</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/f14d98f689925e2d93cad95d5c043dd103238cec/analysis/benchmark.Rmd" target="_blank">f14d98f</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/421d5d44f065639498503f68193015533d86883b/docs/benchmark.html" target="_blank">421d5d4</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/7e810a54c78c0c06c2065e5dac02180c235bca9b/analysis/benchmark.Rmd" target="_blank">7e810a5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
workflowr::wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/ef624c3b4397ed623a0d70d33b1434f26ec558f1/docs/benchmark.html" target="_blank">ef624c3</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/d83b8be0d9604a14b58ba504f3855ac91d668bc2/analysis/benchmark.Rmd" target="_blank">d83b8be</a>
</td>
<td>
schae211
</td>
<td>
2021-12-14
</td>
<td>
workflowr::wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/487e9ef41f3e9db110529e2e87697965ec804547/docs/benchmark.html" target="_blank">487e9ef</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/8fcde1bffefbbabbea7647ae9778e92062513766/analysis/benchmark.Rmd" target="_blank">8fcde1b</a>
</td>
<td>
schae211
</td>
<td>
2021-12-06
</td>
<td>
wflow_publish("analysis/benchmark.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/067350bf869d53ad7ccb91c13d49c77e2d78860d/docs/benchmark.html" target="_blank">067350b</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/b68d23d819a8167d19093d76853ba64c512ca32d/docs/benchmark.html" target="_blank">b68d23d</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/565f4e02511a796a9de49f418982b5a8cb298745/analysis/benchmark.Rmd" target="_blank">565f4e0</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/a450bedf5080c68a7df841975f8f12351c37827f/docs/benchmark.html" target="_blank">a450bed</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/e197adb7350543bca1877b7f0422f9948151f382/analysis/benchmark.Rmd" target="_blank">e197adb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-02
</td>
<td>
wflow_publish(files = "analysis/benchmark.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Setup.</p>
<pre class="r"><code>suppressPackageStartupMessages(library(mistyR))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(magrittr))
suppressPackageStartupMessages(library(slider))</code></pre>
<div id="introduction-and-data" class="section level1 tabset">
<h1 class="tabset">Introduction and Data</h1>
<p>Loading the data.</p>
<pre><code>[1] &quot;Loading results from: output/2021-12-23-09-57_all_results.RDS&quot;</code></pre>
<pre><code>[1] &quot;Loading views from: output/2021-12-17-07-59_all_views.RDS&quot;</code></pre>
<p>What data / experiments is the benchmarking based on?</p>
<pre class="r"><code>str_extract(names(all.results), &quot;[^/]+(?=/)&quot;) %&gt;% unique</code></pre>
<pre><code>[1] &quot;4iANCA&quot;           &quot;merfish_bc&quot;       &quot;merfish_liver&quot;    &quot;merfish_preoptic&quot;
[5] &quot;mibi_tnbc&quot;        &quot;synthetic&quot;       </code></pre>
<div id="synthetic" class="section level2">
<h2><code>synthetic</code></h2>
<ul>
<li><p>Simulated data that are included in the MISTy package. As explained in the <a href="https://saezlab.github.io/mistyR/articles/mistyR.html">Get Started Vignette</a>,this dataset is based on a two-dimensional cellular automata model which models the production, diffusion, degradation and interactions of 11 molecular species in 4 different cell types.</p></li>
<li><p>Data Availability: <code>mistyR</code> package <code>data("synthetic")</code></p></li>
<li><p>Features: Simulated Data</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 10</code></pre>
<ul>
<li>Number of markers in each sample:</li>
</ul>
<pre><code>[1] 11</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><p><code>/synthetic/l6</code>: paraview computed with <code>l=6</code></p></li>
<li><p><code>/synthetic/l12</code>: paraview computed with <code>l=12</code></p></li>
<li><p><code>/synthetic/l24</code>: paraview computed with <code>l=24</code></p></li>
</ul></li>
</ul>
<pre><code>[1] &quot;/synthetic/l12&quot; &quot;/synthetic/l24&quot; &quot;/synthetic/l6&quot; </code></pre>
<ul>
<li>Number of spatial units (e.g. cells) per sample:</li>
</ul>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   4110    4120    4148    4152    4179    4205 </code></pre>
<hr />
</div>
<div id="mibi_tnbc" class="section level2">
<h2><code>mibi_tnbc</code></h2>
<ul>
<li><p>Multiplexed Ion Beam Imaging (spatial proteomics) dataset from <a href="https://dx.doi.org/10.1016/j.cell.2018.08.039">A Structured Tumor-Immune Microenvironment in Triple Negative Breast Cancer Revealed by Multiplexed Ion Beam Imaging</a> by Keren, L. et al. (Cell 174, 1373-1387.e19 (2018)) <span class="citation">(<em>1</em>)</span></p></li>
<li><p>Data Availability: <a href="https://mibi-share.ionpath.com/" class="uri">https://mibi-share.ionpath.com/</a></p></li>
<li><p>Features: Spatial Proteomics, Single Cell Resolution, Segmentation</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 40</code></pre>
<ul>
<li>Number of markers in each sample:</li>
</ul>
<pre><code>[1] 28</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><code>standard_views</code>: markers with zero variance were removed from each sample, juxtaview was computed with <code>neighbor.thr = 40</code>, and paraview was computed with <code>l = 120, zoi = 40</code>.</li>
</ul></li>
</ul>
<pre><code>[1] &quot;/mibi_tnbc/standard_views&quot;</code></pre>
<ul>
<li>Number of spatial units (e.g. cells) per sample:</li>
</ul>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2046    3876    4961    4942    6080    8212 </code></pre>
<hr />
</div>
<div id="merfish_preoptic" class="section level2">
<h2><code>merfish_preoptic</code></h2>
<ul>
<li><p>MERFISH data from the murine hypothalamic preoptic region from <a href="https://dx.doi.org/10.1126/science.aau5324">Molecular, spatial, and functional single-cell profiling of the hypothalamic preoptic region</a> by Moffitt, R. et al. (Science 362, 6416 (2018))</p></li>
<li><p>Data Availability: <a href="http://www.spatialomics.org/SpatialDB/merfish_30385464_browse.php">Spatial DB</a></p></li>
<li><p>Features: Spatial Transcriptomics, Single Cell Resolution, Segmentation</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 36</code></pre>
<ul>
<li>Number of markers in each sample (depending on the view, see explanation below):</li>
</ul>
<pre><code>[1] 108 155</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><p><code>standard_views</code>: markers with zero variance were removed from each sample, the juxtaview was computed with <code>neighbor.thr = 30</code>, and the paraview with <code>l = 120, zoi = 30</code>.</p></li>
<li><p><code>hvg_views</code>: markers with zero variance were removed from each sample, the 108 most spatially variable genes (computed with <em>SPARK-X</em> <span class="citation">(<em>2</em>)</span>) were selected, the juxtaview was computed with <code>neighbor.thr = 30</code>, and the paraview with <code>l = 120, zoi = 30</code>.</p></li>
</ul></li>
</ul>
<pre><code>[1] &quot;/merfish_preoptic/hvg_views&quot;      &quot;/merfish_preoptic/standard_views&quot;</code></pre>
<ul>
<li>Number of spatial units (e.g. cells) per sample:</li>
</ul>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  10291   21938   22642   28551   32065   73655 </code></pre>
<hr />
</div>
<div id="merfish_liver" class="section level2">
<h2><code>merfish_liver</code></h2>
<ul>
<li><p>MERFISH data from the murine fetal liver from <a href="https://dx.doi.org/10.1038/s41421-021-00266-1">Spatial transcriptome profiling by MERFISH reveals fetal liver hematopoietic stem cell niche architecture</a> by Yanfang, L. et al. (Cell Discovery 7, 47 (2021))</p></li>
<li><p>Data Availability: <a href="https://campuspress.yale.edu/wanglab/HSCMERFISH/">SIYUAN WANG LAB</a></p></li>
<li><p>Features: Spatial Transcriptomics, Single Cell Resolution, Segmentation</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 164</code></pre>
<ul>
<li>Number of markers in each sample:</li>
</ul>
<pre><code>[1] 140</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><code>standard280_views</code>: markers with zero variance were removed from each sample, samples with fewer than 280 cells were removed, the juxtaview was computed with <code>neighbor.thr = 100</code>, and the paraview with <code>l = 180, zoi = 100</code>.</li>
</ul></li>
</ul>
<pre><code>[1] &quot;/merfish_liver/standard280_views&quot;</code></pre>
<ul>
<li>Number of spatial units (e.g. cells) per sample:</li>
</ul>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  280.0   296.8   315.0   315.4   334.0   366.0 </code></pre>
<hr />
</div>
<div id="merfish_bc" class="section level2">
<h2><code>merfish_bc</code></h2>
<ul>
<li><p>MERFISH data from metastatic breast cancer (unpublished)</p></li>
<li><p>Data Availability: -</p></li>
<li><p>Features: Spatial Transcriptomics, Single Cell Resolution, Segmentation</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 9</code></pre>
<ul>
<li>Number of markers in each sample (depending on the view, see explanation below):</li>
</ul>
<pre><code>[1]  86 291</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><p><code>standard_views</code>: markers with zero variance were removed from each sample, the juxtaview was computed with <code>neighbor.thr = 15</code>, and the paraview with <code>l = 100, zoi = 15</code>.</p></li>
<li><p><code>hvg_views</code>: markers with zero variance were removed from each sample, the 86 most spatially variable genes (computed with <em>SPARK-X</em> <span class="citation">(<em>2</em>)</span>) were selected, the juxtaview was computed with <code>neighbor.thr = 15</code>, and the paraview with <code>l = 100, zoi = 15</code>.</p></li>
</ul></li>
</ul>
<pre><code>[1] &quot;/merfish_bc/hvg_views&quot;      &quot;/merfish_bc/standard_views&quot;</code></pre>
<ul>
<li>Number of spatial units (e.g. cells) per sample:</li>
</ul>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   2704    3981   11014   11787   13822   31197 </code></pre>
<hr />
</div>
<div id="ianca" class="section level2">
<h2><code>4iANCA</code></h2>
<ul>
<li><p>Iiterative Indirect Immunofluorescence Imaging (4i) data from glomeruli (unpublished)</p></li>
<li><p>Data Availability: -</p></li>
<li><p>Features: Spatial Proteomics, Single Cell Resolution, No Segmentation</p></li>
<li><p>Number of samples:</p></li>
</ul>
<pre><code>[1] 36</code></pre>
<ul>
<li>Number of markers in each sample:</li>
</ul>
<pre><code>[1] 27</code></pre>
<ul>
<li><p>Different views that were computed:</p>
<ul>
<li><p><code>/4iANCA/ratio_153_views</code></p></li>
<li><p><code>/4iANCA/ratio_76_views</code></p></li>
<li><p><code>/4iANCA/ratio_38_views</code></p></li>
<li><p><code>/4iANCA/ratio_19_views</code></p></li>
<li><p><code>/4iANCA/ratio_9\_views</code></p></li>
</ul></li>
</ul>
<pre><code>[1] &quot;/4iANCA/ratio_153_views&quot; &quot;/4iANCA/ratio_19_views&quot; 
[3] &quot;/4iANCA/ratio_38_views&quot;  &quot;/4iANCA/ratio_76_views&quot; 
[5] &quot;/4iANCA/ratio_9_views&quot;  </code></pre>
<ul>
<li><p>Number of spatial units (e.g. cells) per sample, depending on the computed view:</p>
<ul>
<li>Number of cells per sample: No segmentation was performed, therefore the pixels were simply binned with 5 different schemes. The pixels have a side length of <span class="math inline">\(0.13\)</span> microns, and an average eukaryotic cell is between <span class="math inline">\(10\)</span> and <span class="math inline">\(100 \micro m\)</span> in diameter. Therefore I choose the largest binning to be <code>floor(20/0.13) = 153</code>, meaning the side length of one spatial unit (bin) will correspond to 153 pixels. The smaller bins are then sequentially half of the size (76, 38, 19, 9). The corresponding parameter for the computation of the paraview was adjusted accordingly, meaning if the l parameter <span class="math inline">\(l_{153}\)</span> was used for computation of the paraview for the 153 binning, <span class="math inline">\(2 \times l_{153}\)</span> was used for the 76 binning.</li>
</ul></li>
</ul>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["experiment"],"name":[1],"type":["chr"],"align":["left"]},{"label":["mean_n"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"/4iANCA/ratio_153_views","2":"169"},{"1":"/4iANCA/ratio_76_views","2":"676"},{"1":"/4iANCA/ratio_38_views","2":"2704"},{"1":"/4iANCA/ratio_19_views","2":"10609"},{"1":"/4iANCA/ratio_9_views","2":"46656"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<hr />
</div>
</div>
<div id="scripts" class="section level1">
<h1>Scripts</h1>
<p>The scripts that were used for all analysis can be found in this <a href="https://github.com/schae211/report/tree/master/cluster_scripts">GitHub repository</a>. The computations were performed on a cluster managed by Slurm.</p>
<hr />
</div>
<div id="comparing-performances" class="section level1 tabset">
<h1 class="tabset">Comparing Performances</h1>
<p>Define a function to get clean data for a particular study.</p>
<pre class="r"><code># helper function to get all misty runs for a particular study
filter_runs &lt;- function(all.runs, str.study, views) {
  str.view &lt;- paste(views, collapse=&quot;|&quot;)
  all.names &lt;- names(all.runs)
  names.study &lt;- all.names[str_detect(all.names, str.study)]
  names.view &lt;- names.study[str_detect(names.study, str.view)]
  sucessful.runs &lt;- map_lgl(all.results[names.view], ~ typeof(.x) == &quot;list&quot;)
  all.results[names.view][sucessful.runs]
}</code></pre>
<p>Define a function to extract the improvements (performance measures) from a list of misty results.</p>
<pre class="r"><code>get_performance &lt;- function(filtered.run) {
  map2_dfr(filtered.run, names(filtered.run),
                                   function(misty.run, name) {
    misty.run$improvements %&gt;%
      mutate(algorithm = str_extract(name, &quot;(?&lt;=/)[^/]+$&quot;)) %&gt;%
      mutate(study = str_extract(sample, &quot;(?&lt;=OUTPUT/)[^/]+&quot;)) %&gt;%
      mutate(view = str_extract(sample, paste0(&quot;(?&lt;=OUTPUT/&quot;, study, &quot;/)[^/]+&quot;)))
  })
}</code></pre>
<p>Define a function to check whether something is missing in misty performance files.</p>
<pre class="r"><code>check_runs &lt;- function(misty.performances, ordering = NULL) {
  
    levs &lt;- function() {
    if (is.null(ordering)) {
      out &lt;- misty.performances %&gt;% 
        filter(!(algorithm %in% exclude)) %&gt;%
        pull(algorithm) %&gt;%
        unique()
    } else {
      out &lt;- ordering
    }
    out
  }
  
  misty.performances %&gt;%
    select(algorithm, sample, view) %&gt;%
    distinct() %&gt;%
    mutate(algorithm = factor(algorithm, levels = levs())) %&gt;%
    group_by(view, algorithm) %&gt;%
    summarise(sample_number = n(), .groups=&quot;drop&quot;) %&gt;%
    pivot_wider(names_from = &quot;view&quot;, values_from = &quot;sample_number&quot;)
}</code></pre>
<p>Define a function to plot misty performances for all targets and for all considered algorithms (and for all views that were included).</p>
<pre class="r"><code>plot_performance &lt;- function(misty.performances, str.measure, 
                             str.summary = &quot;mean&quot;, targets = FALSE,
                             exclude = c(), wrap.views = FALSE,
                             ncol = 2, ordering = NULL) {
  
  str.summary = match.arg(str.summary, c(&quot;mean&quot;, &quot;median&quot;))
  
  fill = ifelse(str.summary == &quot;mean&quot;, &quot;mean&quot;, &quot;median&quot;)
  
  theme_custom &lt;- function() {
    if (targets) {
      out &lt;- theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5))
    } else {
      out &lt;- theme(axis.text.y = element_blank(), axis.ticks.y=element_blank(),
                   axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5))
    }
    out
  }
  
  wrap &lt;- function() {
    if (wrap.views) {
      out &lt;- facet_wrap(~ view, ncol = ncol)
    } else {
      out &lt;- NULL
    }
    out
  }
  
  levs &lt;- function() {
    if (is.null(ordering)) {
      out &lt;- misty.performances %&gt;% 
        filter(!(algorithm %in% exclude)) %&gt;%
        pull(algorithm) %&gt;%
        unique()
    } else {
      out &lt;- ordering
    }
    out
  }
  
  plot &lt;- misty.performances %&gt;%
    filter(measure == str.measure) %&gt;%
    filter(!(algorithm %in% exclude)) %&gt;%
    mutate(algorithm = factor(algorithm, levels=levs())) %&gt;%
    group_by(view, algorithm, target) %&gt;%
    summarise(mean = mean(value), median = median(value), 
              sd = sd(value), .groups = &quot;drop_last&quot;) %&gt;%
    ggplot() +
    geom_tile(aes(x = algorithm, y = target, 
                  # use .data[[var]] to access data variables with strings
                  fill = .data[[fill]] )) +
    scale_fill_viridis_c(values = c(0, 0.1, 0.2, 0.4, 0.7, 1)) +
    labs(x = &quot;Algorithm&quot;, fill = paste0(str.summary, &quot; : &quot;, str.measure)) +
    theme_custom() +
    wrap()
  
  print(plot)
}</code></pre>
<pre class="r"><code># default ordering of the algorithms
def.ord &lt;-  c(&quot;RF&quot;, &quot;TBOOST&quot;, &quot;MARS100&quot;, &quot;MARS&quot;, &quot;MARS80&quot;, &quot;MARS60&quot;, &quot;MARS40&quot;,
              &quot;BGMARS&quot;, &quot;LBOOST&quot;, &quot;LM&quot;, &quot;SVM&quot;, &quot;MLP&quot;)</code></pre>
<div id="synthetic-1" class="section level2">
<h2><code>synthetic</code></h2>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, c(&quot;l6&quot;, &quot;l12&quot;, &quot;l24&quot;)) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["l12"],"name":[2],"type":["int"],"align":["right"]},{"label":["l24"],"name":[3],"type":["int"],"align":["right"]},{"label":["l6"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"RF","2":"10","3":"10","4":"10"},{"1":"TBOOST","2":"10","3":"10","4":"10"},{"1":"MARS","2":"10","3":"10","4":"10"},{"1":"BGMARS","2":"10","3":"10","4":"10"},{"1":"LBOOST","2":"10","3":"10","4":"10"},{"1":"LM","2":"10","3":"10","4":"10"},{"1":"SVM","2":"10","3":"10","4":"10"},{"1":"MLP","2":"10","3":"10","4":"10"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, c(&quot;l6&quot;, &quot;l12&quot;, &quot;l24&quot;)) %&gt;%
  get_performance() %&gt;%
  mutate(view = factor(view, levels = c(&quot;l6&quot;, &quot;l12&quot;, &quot;l24&quot;))) %T&gt;%
  plot_performance(str.measure = &quot;multi.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = TRUE, wrap.views = TRUE, ncol=3,
                   ordering = def.ord) %T&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = TRUE, wrap.views = TRUE, ncol=3,
                   ordering = def.ord) %&gt;%
  invisible()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-35-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-35-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/f72d59c3b66e424cd5f45999439ddcddcf91bbfc/docs/figure/benchmark.Rmd/unnamed-chunk-35-1.png" target="_blank">f72d59c</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-35-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-2">
Past versions of unnamed-chunk-35-2.png
</button>
</p>
<div id="fig-unnamed-chunk-35-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-35-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mibi_tnbc-1" class="section level2">
<h2><code>mibi_tnbc</code></h2>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;standard&quot;) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["standard"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"RF","2":"40"},{"1":"TBOOST","2":"40"},{"1":"MARS","2":"40"},{"1":"BGMARS","2":"40"},{"1":"LBOOST","2":"40"},{"1":"LM","2":"40"},{"1":"SVM","2":"40"},{"1":"MLP","2":"40"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;standard&quot;) %&gt;%
  get_performance() %T&gt;%
  plot_performance(str.measure = &quot;multi.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = TRUE, ordering = def.ord) %T&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = TRUE, ordering = def.ord) %&gt;%
  invisible()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-37-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-37-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-37-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-37-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-37-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/f72d59c3b66e424cd5f45999439ddcddcf91bbfc/docs/figure/benchmark.Rmd/unnamed-chunk-37-1.png" target="_blank">f72d59c</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-37-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-2">
Past versions of unnamed-chunk-37-2.png
</button>
</p>
<div id="fig-unnamed-chunk-37-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-37-2.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-37-2.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-37-2.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-37-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="merfish_preoptic-1" class="section level2">
<h2><code>merfish_preoptic</code></h2>
<p>We see that MARS (with the current configuration) is not scalable enough to handle such large datasets with 100 markers and up to 70,000 spatial units.</p>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_preoptic&quot;, &quot;hvg&quot;) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["hvg"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"RF","2":"35"},{"1":"TBOOST","2":"36"},{"1":"MARS","2":"1"},{"1":"MARS80","2":"1"},{"1":"MARS60","2":"1"},{"1":"MARS40","2":"1"},{"1":"LBOOST","2":"36"},{"1":"LM","2":"36"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_preoptic&quot;, &quot;hvg&quot;) %&gt;%
  get_performance() %T&gt;%
  plot_performance(str.measure = &quot;multi.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, ordering = def.ord,
                   exclude = c(&quot;MARS&quot;, &quot;MARS40&quot;, &quot;MARS60&quot;, &quot;MARS80&quot;)) %T&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, ordering = def.ord,
                   exclude = c(&quot;MARS&quot;, &quot;MARS40&quot;, &quot;MARS60&quot;, &quot;MARS80&quot;)) %&gt;%
  invisible()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-39-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-39-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/f72d59c3b66e424cd5f45999439ddcddcf91bbfc/docs/figure/benchmark.Rmd/unnamed-chunk-39-1.png" target="_blank">f72d59c</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-39-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-2">
Past versions of unnamed-chunk-39-2.png
</button>
</p>
<div id="fig-unnamed-chunk-39-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-39-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="merfish_liver-1" class="section level2">
<h2><code>merfish_liver</code></h2>
<p>For some reason everything but bagged MARS, MARS, and gradient boosted trees failed.</p>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_liver&quot;, &quot;std280&quot;) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["std280"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"TBOOST","2":"149"},{"1":"MARS","2":"164"},{"1":"BGMARS","2":"111"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_liver&quot;, &quot;std280&quot;) %&gt;%
  get_performance() %T&gt;%
  plot_performance(str.measure = &quot;mulit.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, ordering = def.ord) %T&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, ordering = def.ord) %&gt;%
  invisible()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-41-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-41-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-41-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-2">
Past versions of unnamed-chunk-41-2.png
</button>
</p>
<div id="fig-unnamed-chunk-41-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-41-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="merfish_bc-1" class="section level2">
<h2><code>merfish_bc</code></h2>
<p>Since the bagged MARS algorithm could only finish the computation for a single sample I must exclude the result. One should also keep in mind that the MARS algorithm was not able to finish (2 samples are missing).</p>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_bc&quot;, &quot;hvg&quot;) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["hvg"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"RF","2":"9"},{"1":"TBOOST","2":"9"},{"1":"MARS","2":"7"},{"1":"BGMARS","2":"1"},{"1":"LM","2":"9"},{"1":"SVM","2":"2"},{"1":"MLP","2":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_bc&quot;, &quot;hvg&quot;) %&gt;%
  get_performance() %T&gt;%
  plot_performance(str.measure = &quot;multi.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, exclude = c(&quot;BGMARS&quot;),
                   ordering = def.ord) %T&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, str.summary = &quot;mean&quot;,
                   targets = FALSE, exclude = c(&quot;BGMARS&quot;),
                   ordering = def.ord) %&gt;%
  invisible()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-43-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-43-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-43-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-2">
Past versions of unnamed-chunk-43-2.png
</button>
</p>
<div id="fig-unnamed-chunk-43-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-43-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="ianca-1" class="section level2">
<h2><code>4iANCA</code></h2>
<p>The computation for the smallest binning (<code>ratio_9</code>) could not be finished.</p>
<p>Also why do we only have 10 samples instead of 36 samples for ratio 153?</p>
<pre class="r"><code>filter_runs(all.results, &quot;4iANCA&quot;, c(&quot;ratio_153&quot;, &quot;ratio_76&quot;, 
                                     &quot;ratio_38&quot;, &quot;ratio_19&quot;, &quot;ratio_9&quot;)) %&gt;%
  get_performance() %&gt;%
  check_runs(ordering = def.ord)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["fct"],"align":["left"]},{"label":["ratio_153"],"name":[2],"type":["int"],"align":["right"]},{"label":["ratio_19"],"name":[3],"type":["int"],"align":["right"]},{"label":["ratio_38"],"name":[4],"type":["int"],"align":["right"]},{"label":["ratio_76"],"name":[5],"type":["int"],"align":["right"]},{"label":["ratio_9"],"name":[6],"type":["int"],"align":["right"]}],"data":[{"1":"RF","2":"10","3":"36","4":"36","5":"36","6":"36"},{"1":"TBOOST","2":"10","3":"36","4":"36","5":"36","6":"36"},{"1":"MARS100","2":"10","3":"36","4":"36","5":"36","6":"5"},{"1":"MARS80","2":"10","3":"36","4":"36","5":"36","6":"5"},{"1":"MARS60","2":"10","3":"36","4":"36","5":"36","6":"8"},{"1":"BGMARS","2":"10","3":"10","4":"36","5":"36","6":"1"},{"1":"LBOOST","2":"10","3":"36","4":"36","5":"36","6":"36"},{"1":"LM","2":"10","3":"36","4":"36","5":"36","6":"36"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;4iANCA&quot;, c(&quot;ratio_153&quot;, &quot;ratio_76&quot;, &quot;ratio_38&quot;,
                                     &quot;ratio_19&quot;, &quot;ratio_9&quot;)) %&gt;%
  get_performance() %&gt;%
  mutate(view = factor(view, levels=c(&quot;ratio_153&quot;, &quot;ratio_76&quot;, &quot;ratio_38&quot;,
                                     &quot;ratio_19&quot;, &quot;ratio_9&quot;))) %T&gt;%
  plot_performance(str.measure = &quot;multi.R2&quot;, wrap.views = TRUE, ncol=3,
                   ordering = def.ord) %&gt;%
  plot_performance(str.measure = &quot;gain.R2&quot;, wrap.views = TRUE, ncol=3,
                   ordering = def.ord)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-45-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-45-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-45-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-45-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-45-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-2">
Past versions of unnamed-chunk-45-2.png
</button>
</p>
<div id="fig-unnamed-chunk-45-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-45-2.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-45-2.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-45-2.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-45-2.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="performance-figure" class="section level1">
<h1>Performance Figure</h1>
<p>Plot the performances of the different algorithms for all studies in one figure using only one view from each study.</p>
</div>
<div id="comparing-importances" class="section level1 tabset">
<h1 class="tabset">Comparing Importances</h1>
<p>Define function to extract the importances (interaction measures).</p>
<pre class="r"><code>get_importance &lt;- function(filtered.run) {
  map2_dfr(filtered.run, names(filtered.run), function(misty.run, name) {
                                     
    # using str_extract_all drastically improves the performance
    # warning: indices of algorithm, study, and view.comp hardcoded
    all.info &lt;- str_extract_all(misty.run$importances$sample, &quot;(?&lt;=/)[^/]+&quot;)
                                     
    misty.run$importances %&gt;%
      mutate(algorithm = map_chr(all.info, ~ .x[length(.x)-1])) %&gt;%
      mutate(study = map_chr(all.info, ~ .x[length(.x)-3])) %&gt;%
      mutate(view.comp = map_chr(all.info, ~ .x[length(.x)-2]))
  }) %&gt;%
    mutate(sample = str_extract(sample, &quot;[^/]+$&quot;))
}</code></pre>
<p>Define function to check whether all importances are present for each algorithm</p>
<pre class="r"><code>check_algs &lt;- function(misty.performances) {
  misty.performances %&gt;%
    unite(relation, Predictor, Target, sep=&quot;_&quot;) %&gt;%
    select(-c(study, view.comp)) %&gt;%
    drop_na() %&gt;%
    pivot_wider(names_from=relation, values_from=Importance) %&gt;%
    mutate(sample = str_extract(sample, &quot;[^/]+$&quot;)) %&gt;%
    pivot_longer(cols = !c(sample, view, algorithm)) %&gt;%
    mutate(check.NA = is.na(value)) %&gt;%
    group_by(algorithm) %&gt;%
    summarise(n.NAs = sum(check.NA), n = n())
}</code></pre>
<p>Define function to get splits according to view and sample from misty importances.</p>
<pre class="r"><code>split_importances &lt;- function(misty.importances) {
  tidy.df &lt;- misty.importances %&gt;%
    unite(relation, Predictor, Target, sep=&quot;_&quot;) %&gt;%
    drop_na() %&gt;%
    mutate(sample = str_extract(sample, &quot;[^/]+$&quot;)) %&gt;%
    pivot_wider(names_from = algorithm, values_from = Importance) %&gt;%
    replace(is.na(.), 0) %&gt;%
    group_by(view, sample)
  
  list(splits = group_split(tidy.df), keys = group_keys(tidy.df))
}</code></pre>
<p>Define function to compute the cosine similarity given a tibble of vectors, where each column represents a vector. Indices are harcoded.</p>
<pre class="r"><code>cosine_similarity &lt;- function(imp.tbl, relative = &quot;RF&quot;) {
  comp &lt;- imp.tbl %&gt;% pull(relative)
  map_dbl(imp.tbl[-c(1:5)], function(vec) {
    cosine.sim &lt;- (comp %*% vec * 1/(comp %*% comp)**0.5 * 1/(vec %*% vec)**0.5)
  })
}</code></pre>
<p>Define function to compute the jaccard index given a tibble of vectors, where each column represents a vector. Indices are harcoded.</p>
<pre class="r"><code>jaccard_index &lt;- function(imp.tbl, relative = &quot;RF&quot;, cutoff = 0.5) {
  comp &lt;- imp.tbl %&gt;% filter(.data[[relative]] &gt; cutoff) %&gt;% pull(relation)
  algs &lt;- names(imp.tbl)[-c(1:5)] %&gt;% setNames(names(imp.tbl)[-c(1:5)])
  
  map_dbl(algs, function(alg) {
    vec &lt;- imp.tbl %&gt;% filter(.data[[alg]] &gt; cutoff) %&gt;% pull(relation)
    length(intersect(comp, vec)) / length(union(comp, vec))
  })
}</code></pre>
<p>Define wrapper function to compute comparitive statistics (e.g. cosine similarity, jaccard index), given a misty importances tibble.</p>
<p>First, the tibble containing the importances is split according to views and samples. Second, for each sub tibble the</p>
<pre class="r"><code>wrap_function &lt;- function(misty.importances, f, ...) {
  
  ellipsis.args &lt;- list(...)
  # Split the importance tibble corresponding to views and samples
  # since we want to compare the algorithm for the same sample and view
  split.list &lt;- split_importances(misty.importances)
  
  # 1. map over each view
  map_dfr(unique(split.list$keys$view), function(sel.view) {
    
    # 2. Map over each sample corresponding to this view
    split.list$keys %&gt;%
      mutate(id = row_number()) %&gt;%
      filter(view == sel.view) %&gt;%
      pull(id, name = sample) %&gt;%
      map_dfr(function(tbl.id) {
        
        # 3. Compute cosine similarity (and add corresponding view and sample)
        
        args &lt;- rlist::list.merge(list(imp.tbl = split.list$splits[[tbl.id]]), 
                                  ellipsis.args)
        c(do.call(f, args), &quot;id&quot; = tbl.id)
          
      }) %&gt;% mutate(view = sel.view,
                    sample = split.list$keys$sample[as.integer(id)]) 
  }) %&gt;%
    pivot_longer(-c(id, view, sample), 
                 names_to = &quot;algorithm&quot;, values_to = &quot;cosine&quot;) %&gt;%
    
      # 4. Group by algorithm 
      # and get the mean of the cosine similarity for each sample
      group_by(algorithm, view) %&gt;%
      summarise(mean = mean(cosine), sd = sd(cosine), .groups=&quot;drop&quot;) %&gt;%
      arrange(algorithm)
}</code></pre>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  wrap_function(jaccard_index, relative=&quot;RF&quot;, cutoff=0.5)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["view"],"name":[2],"type":["chr"],"align":["left"]},{"label":["mean"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sd"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"BGMARS","2":"intra","3":"0.4885020","4":"0.02765900"},{"1":"BGMARS","2":"para.6","3":"0.5517689","4":"0.05176364"},{"1":"LBOOST","2":"intra","3":"0.0000000","4":"0.00000000"},{"1":"LBOOST","2":"para.6","3":"0.0000000","4":"0.00000000"},{"1":"LM","2":"intra","3":"0.4757976","4":"0.05840807"},{"1":"LM","2":"para.6","3":"0.2286882","4":"0.03718948"},{"1":"MARS","2":"intra","3":"0.4783723","4":"0.03037054"},{"1":"MARS","2":"para.6","3":"0.4836687","4":"0.02722361"},{"1":"MLP","2":"intra","3":"0.3653629","4":"0.02951035"},{"1":"MLP","2":"para.6","3":"0.3090342","4":"0.04172183"},{"1":"RF","2":"intra","3":"1.0000000","4":"0.00000000"},{"1":"RF","2":"para.6","3":"1.0000000","4":"0.00000000"},{"1":"SVM","2":"intra","3":"0.5211930","4":"0.05526075"},{"1":"SVM","2":"para.6","3":"0.2723302","4":"0.03315482"},{"1":"TBOOST","2":"intra","3":"0.4594526","4":"0.02395888"},{"1":"TBOOST","2":"para.6","3":"0.7003634","4":"0.06983127"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Function to compute the cosine similarity for each algorithm for all views and samples.</p>
<pre class="r"><code>cosine_similarity_wrapper &lt;- function(misty.importances, relative = &quot;RF&quot;) {
  
  # Split the importance tibble corresponding to views and samples
  # since we want to compare the algorithm for the same sample and view
  split.list &lt;- split_importances(misty.importances)
  
  # 1. map over each view
  map_dfr(unique(split.list$keys$view), function(sel.view) {
    
    # 2. Map over each sample corresponding to this view
    split.list$keys %&gt;%
      mutate(id = row_number()) %&gt;%
      filter(view == sel.view) %&gt;%
      pull(id, name = sample) %&gt;%
      map_dfr(function(tbl.id) {
        
        # 3. Compute cosine similarity (and add corresponding view and sample)
        c(cosine_similarity(split.list$splits[[tbl.id]], relative), &quot;id&quot; = tbl.id)
          
      }) %&gt;% mutate(view = sel.view,
                    sample = split.list$keys$sample[as.integer(id)]) 
  }) %&gt;%
    pivot_longer(-c(id, view, sample), 
                 names_to = &quot;algorithm&quot;, values_to = &quot;cosine&quot;) %&gt;%
    
      # 4. Group by algorithm 
      # and get the mean of the cosine similarity for each sample
      group_by(algorithm, view) %&gt;%
      summarise(mean = mean(cosine), sd = sd(cosine), .groups=&quot;drop&quot;) %&gt;%
      arrange(algorithm)
}</code></pre>
<pre class="r"><code>plot_cosine_similarity &lt;- function(cosine_similarity) {
  cosine_similarity %&gt;%
    mutate(low = mean-sd, high = mean+sd) %&gt;%
    mutate(algorithm = factor(algorithm, levels = def.ord)) %&gt;%
    ggplot() +
    geom_point(aes(x = algorithm, y = mean)) +
    geom_errorbar(aes(x = algorithm, ymin = low, ymax = high)) +
    facet_wrap(~ view) +
    theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5)) +
    lims(y = c(0, 1))
}</code></pre>
<pre class="r"><code>jaccard_index_wrapper &lt;- function(misty.importances, relative = &quot;RF&quot;, cutoff = 0.5) {
  
  # Split the importance tibble corresponding to views and samples
  # since we want to compare the algorithm for the same sample and view
  split.list &lt;- split_importances(misty.importances)
  
  # 1. map over each view
  map_dfr(unique(split.list$keys$view), function(sel.view) {
    
    # 2. Map over each sample corresponding to this view
    split.list$keys %&gt;%
      mutate(id = row_number()) %&gt;%
      filter(view == sel.view) %&gt;%
      pull(id, name = sample) %&gt;%
      map_dfr(function(tbl.id) {
        
        # 3. Compute jaccard index (and add corresponding view and sample)
        c(jaccard_index(split.list$splits[[tbl.id]], relative, cutoff), &quot;id&quot; = tbl.id)
          
      }) %&gt;% mutate(view = sel.view,
                    sample = split.list$keys$sample[as.integer(id)]) 
  }) %&gt;%
    pivot_longer(-c(id, view, sample), 
                 names_to = &quot;algorithm&quot;, values_to = &quot;jaccard&quot;) %&gt;%
    
      # 4. Group by algorithm 
      # and get the mean of the cosine similarity for each sample
      group_by(algorithm, view) %&gt;%
      summarise(mean = mean(jaccard), sd = sd(jaccard), .groups=&quot;drop&quot;) %&gt;%
      arrange(algorithm)
}</code></pre>
<pre class="r"><code>plot_jaccard &lt;- function(jaccard) {
  jaccard %&gt;%
    mutate(low = mean-sd, high = mean+sd) %&gt;%
    mutate(algorithm = factor(algorithm, levels = def.ord)) %&gt;%
    ggplot() +
    geom_point(aes(x = algorithm, y = mean)) +
    geom_errorbar(aes(x = algorithm, ymin = low, ymax = high)) +
    facet_wrap(~ view) +
    theme(axis.text.x = element_text(angle=45, vjust=0.5, hjust=0.5)) +
    lims(y = c(0, 1))
}</code></pre>
<div id="synthetic-2" class="section level2">
<h2><code>synthetic</code></h2>
<p>For the boosting functions several importances are not returned?</p>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  check_algs()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.NAs"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"BGMARS","2":"0","3":"2200"},{"1":"LBOOST","2":"0","3":"2200"},{"1":"LM","2":"0","3":"2200"},{"1":"MARS","2":"0","3":"2200"},{"1":"MLP","2":"0","3":"2200"},{"1":"RF","2":"0","3":"2200"},{"1":"SVM","2":"0","3":"2200"},{"1":"TBOOST","2":"174","3":"2200"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="cosine-similarity" class="section level3">
<h3>Cosine Similarity</h3>
<p>Now let’s look at the cosine similarities.</p>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["view"],"name":[2],"type":["chr"],"align":["left"]},{"label":["mean"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sd"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"BGMARS","2":"intra","3":"0.80010059","4":"1.847419e-02"},{"1":"BGMARS","2":"para.6","3":"0.70034924","4":"1.690683e-02"},{"1":"LBOOST","2":"intra","3":"NaN","4":"NA"},{"1":"LBOOST","2":"para.6","3":"NaN","4":"NA"},{"1":"LM","2":"intra","3":"0.59308472","4":"1.852187e-02"},{"1":"LM","2":"para.6","3":"-0.01947042","4":"3.892767e-02"},{"1":"MARS","2":"intra","3":"0.77028676","4":"1.825054e-02"},{"1":"MARS","2":"para.6","3":"0.58431604","4":"3.521139e-02"},{"1":"MLP","2":"intra","3":"0.64228440","4":"2.369736e-02"},{"1":"MLP","2":"para.6","3":"0.42610378","4":"2.435441e-02"},{"1":"RF","2":"intra","3":"1.00000000","4":"1.812987e-16"},{"1":"RF","2":"para.6","3":"1.00000000","4":"1.958249e-16"},{"1":"SVM","2":"intra","3":"0.70216341","4":"1.714482e-02"},{"1":"SVM","2":"para.6","3":"0.18976885","4":"3.194078e-02"},{"1":"TBOOST","2":"intra","3":"0.76576572","4":"1.948244e-02"},{"1":"TBOOST","2":"para.6","3":"0.89473729","4":"2.046321e-02"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>And plot them.</p>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper() %&gt;%
  plot_cosine_similarity()</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-60-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-60-1">
Past versions of unnamed-chunk-60-1.png
</button>
</p>
<div id="fig-unnamed-chunk-60-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-60-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-60-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="jaccard-index-after-binarization" class="section level3">
<h3>Jaccard Index after Binarization</h3>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  jaccard_index_wrapper(cutoff = 0) %&gt;%
  plot_jaccard()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-61-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-61-1">
Past versions of unnamed-chunk-61-1.png
</button>
</p>
<div id="fig-unnamed-chunk-61-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-61-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-61-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pearson-correlation" class="section level3">
<h3>Pearson Correlation</h3>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  unite(relation, Predictor, Target) %&gt;%
  drop_na() %&gt;%
  pivot_wider(names_from = relation, values_from = Importance) %&gt;%
  filter(view == &quot;intra&quot;) %&gt;%
  select(-c(study, view.comp)) %&gt;%
  mutate(sample = str_extract(sample, &quot;[^/]+$&quot;)) %&gt;%
  group_by(algorithm) %&gt;%
  summarise(across(4:112, mean)) %&gt;%
  replace(is.na(.), 0) %&gt;%
  pivot_longer(cols = !algorithm) %&gt;%
  pivot_wider(names_from = algorithm, values_from = value) %&gt;%
  select(-name) %&gt;%
  cor() %&gt;% 
  corrplot::corrplot(method=&quot;color&quot;, col.lim = c(0,1), 
      col = colorRampPalette(c(&quot;#414487FF&quot;, &quot;#414487FF&quot;, &quot;#414487FF&quot;,
                               &quot;#414487FF&quot;, &quot;#22A884FF&quot;, &quot;#7AD151FF&quot;,
                               &quot;#FDE725FF&quot;))(100),
      tl.col = &quot;black&quot;)</code></pre>
<pre><code>Warning in cor(.): the standard deviation is zero</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-62-1">
Past versions of unnamed-chunk-62-1.png
</button>
</p>
<div id="fig-unnamed-chunk-62-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-62-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-62-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/e9c007647ea473f2118b7e74517932deb1b93556/docs/figure/benchmark.Rmd/unnamed-chunk-62-1.png" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>tidy.df &lt;- filter_runs(all.results, &quot;synthetic&quot;, &quot;l6&quot;) %&gt;%
  get_importance() %&gt;%
  unite(relation, Predictor, Target) %&gt;%
  drop_na() %&gt;%
  pivot_wider(names_from = relation, values_from = Importance) %&gt;%
  filter(view == &quot;intra&quot;) %&gt;%
  select(-c(study, view.comp)) %&gt;%
  mutate(sample = str_extract(sample, &quot;[^/]+$&quot;)) %&gt;%
  group_by(algorithm) %&gt;%
  summarise(across(4:112, mean)) %&gt;%
  replace(is.na(.), 0) %&gt;%
  pivot_longer(cols = !algorithm) %&gt;%
  pivot_wider(names_from = algorithm, values_from = value) %&gt;%
  rename(importance = name) %&gt;%
  pivot_longer(cols = !c(importance, RF)) %&gt;%
  mutate(name = factor(name))

ggplot() +
  geom_point(aes(x = tidy.df$RF, y = tidy.df$value, col = tidy.df$name)) +
  geom_line(aes(x = tidy.df$RF, y = tidy.df$RF), col = &quot;blue&quot;) +
  coord_fixed() +
  scale_color_brewer(palette=&quot;Dark2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-63-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-63-1">
Past versions of unnamed-chunk-63-1.png
</button>
</p>
<div id="fig-unnamed-chunk-63-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-63-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-63-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/e9c007647ea473f2118b7e74517932deb1b93556/docs/figure/benchmark.Rmd/unnamed-chunk-63-1.png" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="mibi_tnbc-2" class="section level2">
<h2><code>mibi_tnbc</code></h2>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;standard&quot;) %&gt;%
  get_importance() %&gt;%
  check_algs()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.NAs"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"BGMARS","2":"34170","3":"151200"},{"1":"LBOOST","2":"34170","3":"151200"},{"1":"LM","2":"34170","3":"151200"},{"1":"MARS","2":"34170","3":"151200"},{"1":"MLP","2":"34170","3":"151200"},{"1":"RF","2":"34170","3":"151200"},{"1":"SVM","2":"34566","3":"151200"},{"1":"TBOOST","2":"47018","3":"151200"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="cosine-similarity-1" class="section level3">
<h3>Cosine Similarity</h3>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;standard&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper() %&gt;%
  plot_cosine_similarity()</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-65-1">
Past versions of unnamed-chunk-65-1.png
</button>
</p>
<div id="fig-unnamed-chunk-65-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-65-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-65-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/e9c007647ea473f2118b7e74517932deb1b93556/docs/figure/benchmark.Rmd/unnamed-chunk-65-1.png" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-65-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="merfish_preoptic-2" class="section level2">
<h2><code>merfish_preoptic</code></h2>
<div id="cosine-similarity-2" class="section level3">
<h3>Cosine Similarity</h3>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_preoptic&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  check_algs()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.NAs"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"LBOOST","2":"0","3":"1248048"},{"1":"LM","2":"0","3":"1248048"},{"1":"MARS","2":"33705","3":"34668"},{"1":"MARS40","2":"28569","3":"34668"},{"1":"MARS60","2":"32742","3":"34668"},{"1":"MARS80","2":"33705","3":"34668"},{"1":"RF","2":"0","3":"1213380"},{"1":"TBOOST","2":"142643","3":"1248048"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_preoptic&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper() %&gt;%
  plot_cosine_similarity()</code></pre>
<pre><code>Warning: Removed 24 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-67-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-67-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/e9c007647ea473f2118b7e74517932deb1b93556/docs/figure/benchmark.Rmd/unnamed-chunk-67-1.png" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-67-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_preoptic&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  jaccard_index_wrapper() %&gt;%
  plot_jaccard()</code></pre>
<pre><code>Warning: Removed 6 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-68-1">
Past versions of unnamed-chunk-68-1.png
</button>
</p>
<div id="fig-unnamed-chunk-68-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-68-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/e9c007647ea473f2118b7e74517932deb1b93556/docs/figure/benchmark.Rmd/unnamed-chunk-68-1.png" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/3a864d203ceeb495196ea22f94cfdcc8c7060e0d/docs/figure/benchmark.Rmd/unnamed-chunk-68-1.png" target="_blank">3a864d2</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="merfish_liver-2" class="section level2">
<h2><code>merfish_liver</code></h2>
</div>
<div id="merfish_bc-2" class="section level2">
<h2><code>merfish_bc</code></h2>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_bc&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  check_algs()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.NAs"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"BGMARS","2":"9435","3":"21930"},{"1":"LM","2":"28815","3":"197370"},{"1":"MARS","2":"12495","3":"153510"},{"1":"MLP","2":"2040","3":"43860"},{"1":"RF","2":"10710","3":"197370"},{"1":"SVM","2":"14790","3":"43860"},{"1":"TBOOST","2":"48335","3":"197370"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="cosine-similarity-3" class="section level3">
<h3>Cosine Similarity</h3>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_bc&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper() %&gt;%
  plot_cosine_similarity()</code></pre>
<pre><code>Warning: Removed 12 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-70-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-70-1">
Past versions of unnamed-chunk-70-1.png
</button>
</p>
<div id="fig-unnamed-chunk-70-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-70-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-70-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;merfish_bc&quot;, &quot;hvg&quot;) %&gt;%
  get_importance() %&gt;%
  jaccard_index_wrapper() %&gt;%
  plot_jaccard()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-71-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-71-1">
Past versions of unnamed-chunk-71-1.png
</button>
</p>
<div id="fig-unnamed-chunk-71-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-71-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="ianca-2" class="section level2">
<h2><code>4iANCA</code></h2>
<pre class="r"><code>filter_runs(all.results, &quot;4iANCA&quot;, &quot;ratio_153&quot;) %&gt;%
  get_importance() %&gt;%
  check_algs()</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["algorithm"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.NAs"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"BGMARS","2":"0","3":"14040"},{"1":"LBOOST","2":"1592","3":"14040"},{"1":"LM","2":"0","3":"14040"},{"1":"MARS100","2":"0","3":"14040"},{"1":"MARS60","2":"0","3":"14040"},{"1":"MARS80","2":"0","3":"14040"},{"1":"RF","2":"0","3":"14040"},{"1":"TBOOST","2":"1592","3":"14040"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="cosine-similarity-4" class="section level3">
<h3>Cosine Similarity</h3>
<pre class="r"><code>filter_runs(all.results, &quot;4iANCA&quot;, &quot;ratio_153&quot;) %&gt;%
  get_importance() %&gt;%
  cosine_similarity_wrapper() %&gt;%
  plot_cosine_similarity()</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-73-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-73-1">
Past versions of unnamed-chunk-73-1.png
</button>
</p>
<div id="fig-unnamed-chunk-73-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-73-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-73-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;4iANCA&quot;, &quot;ratio_153&quot;) %&gt;%
  get_importance() %&gt;%
  jaccard_index_wrapper() %&gt;%
  plot_jaccard()</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-74-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-74-1">
Past versions of unnamed-chunk-74-1.png
</button>
</p>
<div id="fig-unnamed-chunk-74-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-74-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-74-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="comparing-runtimes" class="section level1">
<h1>Comparing Runtimes</h1>
<p>Small timing benchmark based on the whole <code>synthetic</code> dataset, using the view composition: intraview and paraview with <code>l = 12</code> and executing the code with 12 workers.</p>
<pre class="r"><code>suppressPackageStartupMessages(library(mistyR))
suppressPackageStartupMessages(library(future))
plan(&quot;multisession&quot;, workers=12)

c(&quot;RF&quot; = random_forest_model, &quot;TBOOST&quot; = gradient_boosting_model, 
  &quot;MARS&quot; = mars_model, &quot;BGMARS&quot; = bagged_mars_model,
  &quot;LM&quot; = linear_model, &quot;SVM&quot; = svm_model, &quot;MLP&quot; = mlp_model) %&gt;%
  imap_dfr(function(fun, fun.name) {
    start &lt;- Sys.time()
    misty.results.path &lt;- 
      imap_chr(all.views$`/synthetic/l12`, function(misty.views, name) {
        misty.views %&gt;%
          run_misty(model.function = fun, results.folder = paste0(&quot;results/&quot;, name))
    })
    end &lt;- Sys.time()
    unlink(&quot;results&quot;, recursive = TRUE)
    tibble::tibble(model.function = fun.name, time = (end - start))
  }) %&gt;%
  cbind(&quot;relative_to_RF&quot; = as.double(.$time) / 
          as.double(rep(.$time[.$model.function==&quot;RF&quot;], length(.$model.function)))) %&gt;%
  arrange(time)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["model.function"],"name":[1],"type":["chr"],"align":["left"]},{"label":["time"],"name":[2],"type":["drtn"],"align":["right"]},{"label":["relative_to_RF"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"LM","2":"38.97360 secs","3":"0.4887075"},{"1":"RF","2":"79.74833 secs","3":"1.0000000"},{"1":"TBOOST","2":"90.08630 secs","3":"1.1296324"},{"1":"MARS","2":"99.28674 secs","3":"1.2450009"},{"1":"SVM","2":"265.67103 secs","3":"3.3313681"},{"1":"MLP","2":"329.59916 secs","3":"4.1329916"},{"1":"BGMARS","2":"730.47235 secs","3":"9.1597201"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>As expected the linear model is by far the fastest model. Random forest and gradient boosting with trees are equally fast and perform much better than the multi-layer perceptron, bagged mars models, and the support vector machine.</p>
</div>
<div id="hyperparameter-optimization" class="section level1 tabset">
<h1 class="tabset">Hyperparameter Optimization</h1>
<div id="rf" class="section level2">
<h2>RF</h2>
<div id="mibi_tnbc-3" class="section level3">
<h3><code>mibi_tnbc</code></h3>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;RF_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;RF_hyper_&quot;)) %&gt;%
  select(algorithm) %&gt;% 
  distinct() %&gt;%
  mutate(name = algorithm) %&gt;%
  separate(algorithm, 
           c(&quot;n.tree&quot;, &quot;min.node.size&quot;, &quot;max.depth&quot;, &quot;splitrule&quot;), &quot;_&quot;) %&gt;%
  select(name, everything())</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n.tree"],"name":[2],"type":["chr"],"align":["left"]},{"label":["min.node.size"],"name":[3],"type":["chr"],"align":["left"]},{"label":["max.depth"],"name":[4],"type":["chr"],"align":["left"]},{"label":["splitrule"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"100_1_0_variance","2":"100","3":"1","4":"0","5":"variance"},{"1":"100_50_0_variance","2":"100","3":"50","4":"0","5":"variance"},{"1":"100_5_0_extratrees","2":"100","3":"5","4":"0","5":"extratrees"},{"1":"100_5_0_maxstat","2":"100","3":"5","4":"0","5":"maxstat"},{"1":"100_5_0_variance","2":"100","3":"5","4":"0","5":"variance"},{"1":"100_5_10_variance","2":"100","3":"5","4":"10","5":"variance"},{"1":"100_5_1_variance","2":"100","3":"5","4":"1","5":"variance"},{"1":"200_5_0_variance","2":"200","3":"5","4":"0","5":"variance"},{"1":"500_5_0_variance","2":"500","3":"5","4":"0","5":"variance"},{"1":"50_5_0_variance","2":"50","3":"5","4":"0","5":"variance"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;RF_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;RF_hyper_&quot;)) %&gt;%
  plot_performance(str.measure=&quot;multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-77-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-77-1">
Past versions of unnamed-chunk-77-1.png
</button>
</p>
<div id="fig-unnamed-chunk-77-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-77-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-77-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/bbf04bbb5e42a9c22992a5299c8cf0c5fc22b99f/docs/figure/benchmark.Rmd/unnamed-chunk-77-1.png" target="_blank">bbf04bb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-77-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/dc93ff5b67be49a0cf5ee4af2e2782eaae9e0f3a/docs/figure/benchmark.Rmd/unnamed-chunk-77-1.png" target="_blank">dc93ff5</a>
</td>
<td>
schae211
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can clearly see that <code>max.depths = 1</code> reduces the performance significantly.</p>
</div>
</div>
<div id="mars" class="section level2">
<h2>MARS</h2>
<div id="mibi_tnbc-4" class="section level3">
<h3><code>mibi_tnbc</code></h3>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;MARS_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;MARS_hyper_&quot;)) %&gt;%
  select(algorithm) %&gt;% 
  distinct() %&gt;%
  mutate(name = algorithm) %&gt;%
  separate(algorithm, 
           c(&quot;penalty&quot;, &quot;degree&quot;, &quot;fast.k&quot;, &quot;pmethod&quot;), &quot;_&quot;) %&gt;%
  select(name, everything())</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["penalty"],"name":[2],"type":["chr"],"align":["left"]},{"label":["degree"],"name":[3],"type":["chr"],"align":["left"]},{"label":["fast.k"],"name":[4],"type":["chr"],"align":["left"]},{"label":["pmethod"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"-1_1_20_backward","2":"-1","3":"1","4":"20","5":"backward"},{"1":"-1_2_20_backward","2":"-1","3":"2","4":"20","5":"backward"},{"1":"0_1_20_backward","2":"0","3":"1","4":"20","5":"backward"},{"1":"0_2_20_backward","2":"0","3":"2","4":"20","5":"backward"},{"1":"2_1_20_backward","2":"2","3":"1","4":"20","5":"backward"},{"1":"2_2_20_backward","2":"2","3":"2","4":"20","5":"backward"},{"1":"3_2_10_backward","2":"3","3":"2","4":"10","5":"backward"},{"1":"3_2_20_backward","2":"3","3":"2","4":"20","5":"backward"},{"1":"3_2_20_forward","2":"3","3":"2","4":"20","5":"forward"},{"1":"3_2_20_none","2":"3","3":"2","4":"20","5":"none"},{"1":"3_2_5_backward","2":"3","3":"2","4":"5","5":"backward"},{"1":"4_2_20_backward","2":"4","3":"2","4":"20","5":"backward"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;MARS_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;MARS_hyper_&quot;)) %&gt;%
  plot_performance(str.measure=&quot;multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-79-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-79-1">
Past versions of unnamed-chunk-79-1.png
</button>
</p>
<div id="fig-unnamed-chunk-79-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-79-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-79-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/bbf04bbb5e42a9c22992a5299c8cf0c5fc22b99f/docs/figure/benchmark.Rmd/unnamed-chunk-79-1.png" target="_blank">bbf04bb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/781ba638f2acfdc32de627718c184805cc4c9048/docs/figure/benchmark.Rmd/unnamed-chunk-79-1.png" target="_blank">781ba63</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interestingly the performance seems to be pretty much invariant to the choice of hyperparameters here.</p>
</div>
<div id="synthetic-3" class="section level3">
<h3><code>synthetic</code></h3>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;MARS_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;MARS_hyper_&quot;)) %&gt;%
  select(algorithm) %&gt;% 
  distinct() %&gt;%
  mutate(name = algorithm) %&gt;%
  separate(algorithm, 
           c(&quot;penalty&quot;, &quot;degree&quot;, &quot;fast.k&quot;, &quot;pmethod&quot;), &quot;_&quot;) %&gt;%
  select(name, everything())</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["penalty"],"name":[2],"type":["chr"],"align":["left"]},{"label":["degree"],"name":[3],"type":["chr"],"align":["left"]},{"label":["fast.k"],"name":[4],"type":["chr"],"align":["left"]},{"label":["pmethod"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"-1_1_20_backward","2":"-1","3":"1","4":"20","5":"backward"},{"1":"-1_2_20_backward","2":"-1","3":"2","4":"20","5":"backward"},{"1":"0_1_20_backward","2":"0","3":"1","4":"20","5":"backward"},{"1":"0_2_20_backward","2":"0","3":"2","4":"20","5":"backward"},{"1":"2_1_20_backward","2":"2","3":"1","4":"20","5":"backward"},{"1":"2_2_20_backward","2":"2","3":"2","4":"20","5":"backward"},{"1":"3_2_10_backward","2":"3","3":"2","4":"10","5":"backward"},{"1":"3_2_20_backward","2":"3","3":"2","4":"20","5":"backward"},{"1":"3_2_20_forward","2":"3","3":"2","4":"20","5":"forward"},{"1":"3_2_20_none","2":"3","3":"2","4":"20","5":"none"},{"1":"3_2_5_backward","2":"3","3":"2","4":"5","5":"backward"},{"1":"4_2_20_backward","2":"4","3":"2","4":"20","5":"backward"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>filter_runs(all.results, &quot;synthetic&quot;, &quot;MARS_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;MARS_hyper_&quot;)) %&gt;%
  plot_performance(str.measure=&quot;multi.R2&quot;, targets = TRUE)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-81-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-81-1">
Past versions of unnamed-chunk-81-1.png
</button>
</p>
<div id="fig-unnamed-chunk-81-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/bbf04bbb5e42a9c22992a5299c8cf0c5fc22b99f/docs/figure/benchmark.Rmd/unnamed-chunk-81-1.png" target="_blank">bbf04bb</a>
</td>
<td>
schae211
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For the synthetic datasets we clearly see that a higher degree increases the model performance.</p>
<p>What about the timing?</p>
<pre class="r"><code>suppressPackageStartupMessages(library(mistyR))
suppressPackageStartupMessages(library(future))
plan(&quot;multisession&quot;, workers=12)

# Penalties: -1, 0, 2, 3
penalties &lt;- c(-1, 0, 2, -1, 0, 2, 3, 4, 3, 3, 3, 3)

# Degrees: 1, 2, 3
degrees &lt;- c(1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2)

# Fast K: 20, 10, 5
fastks &lt;- c(20, 20, 20, 20, 20, 20, 20, 20, 10, 5, 20, 20)

# Prunning Method: &quot;backward&quot;, &quot;none&quot;, &quot;forward&quot;
pmethods &lt;- c(&quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, 
              &quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, &quot;backward&quot;, 
              &quot;none&quot;, &quot;forward&quot;)

pmap_dfr(list(penalties, degrees, fastks, pmethods),
         function(penalty, degree, fast.k, pmethod) {
           print(paste(penalty, degree, fast.k, pmethod, sep=&quot;_&quot;))
           start &lt;- Sys.time()
           
           walk(all.views$`/synthetic/l12`, function(view) {
             run_misty(views = view, model.function = mars_model,
                       degree = degree, penalty = penalty, 
                       fast.k = fast.k, pmethod = pmethod)
           })
           
           end &lt;- Sys.time()
           unlink(&quot;results&quot;)
           tibble(name = paste(penalty, degree, fast.k, pmethod, sep=&quot;_&quot;),
                  penalty = penalty, degree = degree, fast.k = fast.k, pmethod = pmethod,
                  time = end - start)
         }) %&gt;%
  arrange(time)</code></pre>
<pre><code>[1] &quot;-1_1_20_backward&quot;
[1] &quot;0_1_20_backward&quot;
[1] &quot;2_1_20_backward&quot;
[1] &quot;-1_2_20_backward&quot;
[1] &quot;0_2_20_backward&quot;
[1] &quot;2_2_20_backward&quot;
[1] &quot;3_2_20_backward&quot;
[1] &quot;4_2_20_backward&quot;
[1] &quot;3_2_10_backward&quot;
[1] &quot;3_2_5_backward&quot;
[1] &quot;3_2_20_none&quot;
[1] &quot;3_2_20_forward&quot;</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["penalty"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["degree"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["fast.k"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["pmethod"],"name":[5],"type":["chr"],"align":["left"]},{"label":["time"],"name":[6],"type":["drtn"],"align":["right"]}],"data":[{"1":"2_1_20_backward","2":"2","3":"1","4":"20","5":"backward","6":"57.36026 secs"},{"1":"0_1_20_backward","2":"0","3":"1","4":"20","5":"backward","6":"57.84322 secs"},{"1":"3_2_5_backward","2":"3","3":"2","4":"5","5":"backward","6":"68.05167 secs"},{"1":"-1_1_20_backward","2":"-1","3":"1","4":"20","5":"backward","6":"78.47299 secs"},{"1":"3_2_10_backward","2":"3","3":"2","4":"10","5":"backward","6":"85.67498 secs"},{"1":"-1_2_20_backward","2":"-1","3":"2","4":"20","5":"backward","6":"99.01776 secs"},{"1":"2_2_20_backward","2":"2","3":"2","4":"20","5":"backward","6":"99.07437 secs"},{"1":"0_2_20_backward","2":"0","3":"2","4":"20","5":"backward","6":"99.28172 secs"},{"1":"3_2_20_backward","2":"3","3":"2","4":"20","5":"backward","6":"99.70097 secs"},{"1":"3_2_20_none","2":"3","3":"2","4":"20","5":"none","6":"100.17760 secs"},{"1":"4_2_20_backward","2":"4","3":"2","4":"20","5":"backward","6":"100.98642 secs"},{"1":"3_2_20_forward","2":"3","3":"2","4":"20","5":"forward","6":"101.68177 secs"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="linear-boosting" class="section level2">
<h2>Linear Boosting</h2>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;LBOOST_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;LBOOST_hyper_&quot;)) %&gt;%
  select(algorithm) %&gt;% 
  distinct() %&gt;%
  mutate(name = algorithm) %&gt;%
  separate(algorithm, 
           c(&quot;lambda&quot;, &quot;alpha&quot;), &quot;_&quot;) %&gt;%
  select(name, everything())</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["lambda"],"name":[2],"type":["chr"],"align":["left"]},{"label":["alpha"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"0.5_0","2":"0.5","3":"0"},{"1":"0_0","2":"0","3":"0"},{"1":"0_0.5","2":"0","3":"0.5"},{"1":"0_1","2":"0","3":"1"},{"1":"1_0","2":"1","3":"0"},{"1":"5_0","2":"5","3":"0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="mibi_tnbc-5" class="section level3">
<h3><code>mibi_tnbc</code></h3>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;LBOOST_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;LBOOST_hyper_&quot;)) %&gt;%
  plot_performance(str.measure=&quot;multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-84-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-84-1">
Past versions of unnamed-chunk-84-1.png
</button>
</p>
<div id="fig-unnamed-chunk-84-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-84-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-84-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>No penalty leads to the best results.</p>
</div>
</div>
<div id="tree-boosting" class="section level2">
<h2>Tree Boosting</h2>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;TBOOST_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;TBOOST_hyper_&quot;)) %&gt;%
  select(algorithm) %&gt;% 
  distinct() %&gt;%
  mutate(name = algorithm) %&gt;%
  separate(algorithm, 
           c(&quot;eta&quot;, &quot;max_depth&quot;, &quot;min_child_weight&quot;, &quot;subsample&quot;), &quot;_&quot;) %&gt;%
  select(name, everything())</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["name"],"name":[1],"type":["chr"],"align":["left"]},{"label":["eta"],"name":[2],"type":["chr"],"align":["left"]},{"label":["max_depth"],"name":[3],"type":["chr"],"align":["left"]},{"label":["min_child_weight"],"name":[4],"type":["chr"],"align":["left"]},{"label":["subsample"],"name":[5],"type":["chr"],"align":["left"]}],"data":[{"1":"0.3_12_1_1","2":"0.3","3":"12","4":"1","5":"1"},{"1":"0.3_1_1_1","2":"0.3","3":"1","4":"1","5":"1"},{"1":"0.3_6_1_0.6","2":"0.3","3":"6","4":"1","5":"0.6"},{"1":"0.3_6_1_0.8","2":"0.3","3":"6","4":"1","5":"0.8"},{"1":"0.3_6_1_1","2":"0.3","3":"6","4":"1","5":"1"},{"1":"0.3_6_2_1","2":"0.3","3":"6","4":"2","5":"1"},{"1":"0.3_6_4_1","2":"0.3","3":"6","4":"4","5":"1"},{"1":"0.5_6_1_1","2":"0.5","3":"6","4":"1","5":"1"},{"1":"0.7_6_1_1","2":"0.7","3":"6","4":"1","5":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="mibi_tnbc-6" class="section level3">
<h3><code>mibi_tnbc</code></h3>
<pre class="r"><code>filter_runs(all.results, &quot;mibi_tnbc&quot;, &quot;TBOOST_hyper&quot;) %&gt;% 
  get_performance() %&gt;%
  mutate(algorithm = str_remove(algorithm, &quot;TBOOST_hyper_&quot;)) %&gt;%
  plot_performance(str.measure=&quot;multi.R2&quot;)</code></pre>
<p><img src="figure/benchmark.Rmd/unnamed-chunk-86-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-86-1">
Past versions of unnamed-chunk-86-1.png
</button>
</p>
<div id="fig-unnamed-chunk-86-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/9229052d2535073c992e78722393aa181c209ec7/docs/figure/benchmark.Rmd/unnamed-chunk-86-1.png" target="_blank">9229052</a>
</td>
<td>
schae211
</td>
<td>
2021-12-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/schae211/report/blob/6de2243e92eb53e7a18997952ef54f36a31f40df/docs/figure/benchmark.Rmd/unnamed-chunk-86-1.png" target="_blank">6de2243</a>
</td>
<td>
schae211
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Similarly to the random forest above, we see that <code>max_depth = 1</code> reduces the overall performance.</p>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-10.1016/j.cell.2018.08.039" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">L. Keren, M. Bosse, D. Marquez, R. Angoshtari, S. Jain, S. Varma, S.-R. Yang, A. Kurian, D. V. Valen, R. West, S. C. Bendall, M. Angelo, <a href="https://doi.org/10.1016/j.cell.2018.08.039"><span class="nocase">A Structured Tumor-Immune Microenvironment in Triple Negative Breast Cancer Revealed by Multiplexed Ion Beam Imaging</span></a>. <em>Cell</em>. <strong>174</strong>, 1373–1387.e19 (2018).</div>
</div>
<div id="ref-10.1186/s13059-021-02404-0" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">J. Zhu, S. Sun, X. Zhou, <a href="https://doi.org/10.1186/s13059-021-02404-0"><span class="nocase">SPARK-X: non-parametric modeling enables scalable and robust detection of spatial expression patterns for large spatial transcriptomic studies</span></a>. <em>Genome Biology</em>. <strong>22</strong>, 184 (2021).</div>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /net/data.isilon/ag-saez/bq_pschaefer/SOFTWARE/miniconda3/envs/misty_benchmark/lib/libopenblasp-r0.3.18.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] future_1.23.0   slider_0.2.2    magrittr_2.0.1  forcats_0.5.1  
 [5] stringr_1.4.0   dplyr_1.0.7     purrr_0.3.4     readr_2.1.1    
 [9] tidyr_1.1.4     tibble_3.1.6    ggplot2_3.3.5   tidyverse_1.3.1
[13] mistyR_1.3.3    workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] fs_1.5.0           lubridate_1.8.0    filelock_1.0.2     RColorBrewer_1.1-2
 [5] httr_1.4.2         rprojroot_2.0.2    tools_4.1.1        backports_1.3.0   
 [9] utf8_1.2.2         R6_2.5.1           DBI_1.1.1          colorspace_2.0-2  
[13] withr_2.4.3        tidyselect_1.1.1   processx_3.5.2     compiler_4.1.1    
[17] git2r_0.28.0       cli_3.1.0          rvest_1.0.2        xml2_1.3.2        
[21] labeling_0.4.2     scales_1.1.1       callr_3.7.0        digest_0.6.29     
[25] rmarkdown_2.11     R.utils_2.11.0     pkgconfig_2.0.3    htmltools_0.5.2   
[29] parallelly_1.30.0  dbplyr_2.1.1       fastmap_1.1.0      highr_0.9         
[33] rlang_0.4.12       readxl_1.3.1       rstudioapi_0.13    jquerylib_0.1.4   
[37] farver_2.1.0       generics_0.1.1     jsonlite_1.7.2     R.oo_1.24.0       
[41] rlist_0.4.6.2      Rcpp_1.0.7         munsell_0.5.0      fansi_0.5.0       
[45] furrr_0.2.3        lifecycle_1.0.1    R.methodsS3_1.8.1  stringi_1.7.6     
[49] whisker_0.4        yaml_2.2.1         grid_4.1.1         parallel_4.1.1    
[53] listenv_0.8.0      promises_1.2.0.1   crayon_1.4.2       haven_2.4.3       
[57] hms_1.1.1          knitr_1.36         ps_1.6.0           pillar_1.6.4      
[61] codetools_0.2-18   reprex_2.0.1       glue_1.6.0         evaluate_0.14     
[65] getPass_0.2-2      data.table_1.14.2  modelr_0.1.8       vctrs_0.3.8       
[69] tzdb_0.2.0         httpuv_1.6.4       cellranger_1.1.0   gtable_0.3.0      
[73] assertthat_0.2.1   xfun_0.28          broom_0.7.10       later_1.3.0       
[77] viridisLite_0.4.0  warp_0.2.0         corrplot_0.92      globals_0.14.0    
[81] ellipsis_0.3.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
