<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>1. Introduction</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/viz-1.8.2/viz.js"></script>
<link href="site_libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="site_libs/grViz-binding-1.0.6.1/grViz.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>




<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">TOC</a>
</li>
<li>
  <a href="introduction.html">1. Introduction</a>
</li>
<li>
  <a href="tutorial.html">2. Methods</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    3. Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="benchmark.html">3.1 Benchmark</a>
    </li>
    <li>
      <a href="frameworks.html">3.2 Frameworks</a>
    </li>
    <li>
      <a href="model_api.html">3.3 Alternatives</a>
    </li>
  </ul>
</li>
<li>
  <a href="conclusions.html">4. Conclusions</a>
</li>
<li>
  <a href="appendix.html">5. Appendix</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/schae211/report">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">1. Introduction</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211125code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211125)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211125code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211125)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttreeba40e7e81cc23071afe4a9c98aa1929b9481a260targetblankba40e7ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/schae211/report/tree/ba40e7e81cc23071afe4a9c98aa1929b9481a260" target="_blank">ba40e7e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomschae211reporttreeba40e7e81cc23071afe4a9c98aa1929b9481a260targetblankba40e7ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/schae211/report/tree/ba40e7e81cc23071afe4a9c98aa1929b9481a260" target="_blank">ba40e7e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/2021-12-06-15-42_all_results.RDS
    Ignored:    data/2021-12-08-14-10_all_results.RDS
    Ignored:    data/2021-12-09-09-21_all_results.RDS
    Ignored:    data/2021-12-09-17-26_all_results.RDS
    Ignored:    data/2021-12-13-18-43_all_results.RDS
    Ignored:    data/2021-12-14-10-57_all_views.RDS
    Ignored:    data/2021-12-14-17-59_all_views.RDS
    Ignored:    data/2021-12-14-23-01_all_results.RDS
    Ignored:    data/2021-12-17-07-58_all_results.RDS
    Ignored:    data/2021-12-17-07-59_all_views.RDS
    Ignored:    data/expression.RDS
    Ignored:    data/mibi_tnbc_meta.RDS

Untracked files:
    Untracked:  analysis/case_study_A.Rmd
    Untracked:  cluster_scripts/bash_scripts/call.sh
    Untracked:  cluster_scripts/bash_scripts/call_single.sh
    Untracked:  cluster_scripts/bash_scripts/file.sh
    Untracked:  cluster_scripts/bash_scripts/file_single.sh
    Untracked:  cluster_scripts/bash_scripts/old_scripts/
    Untracked:  cluster_scripts/benchmark/merfish_bc.R
    Untracked:  cluster_scripts/benchmark/merfish_liver.R
    Untracked:  cluster_scripts/benchmark/merfish_preoptic.R
    Untracked:  cluster_scripts/benchmark/mibi_tnbc.R
    Untracked:  code/4i_analysis.R
    Untracked:  code/helper_functions.R
    Untracked:  data/case_study_A/
    Untracked:  output/case_study_B/

Unstaged changes:
    Modified:   analysis/case_study_B.Rmd
    Modified:   analysis/introduction.Rmd
    Deleted:    cluster_scripts/bash_scripts/four_i.sh
    Deleted:    cluster_scripts/bash_scripts/function_caller.sh
    Deleted:    cluster_scripts/bash_scripts/hvg_merfish_bc.sh
    Deleted:    cluster_scripts/bash_scripts/hvg_merfish_liver.sh
    Deleted:    cluster_scripts/bash_scripts/hvg_merfish_preoptic.sh
    Deleted:    cluster_scripts/bash_scripts/lboost_mibi_tnbc.sh
    Deleted:    cluster_scripts/bash_scripts/mars_mibi_tnbc.sh
    Deleted:    cluster_scripts/bash_scripts/mars_synthetic.sh
    Deleted:    cluster_scripts/bash_scripts/rf_mibi_tnbc.sh
    Deleted:    cluster_scripts/bash_scripts/run_four_i_prep.sh
    Deleted:    cluster_scripts/bash_scripts/run_gather_results.sh
    Deleted:    cluster_scripts/bash_scripts/run_prep.sh
    Deleted:    cluster_scripts/bash_scripts/run_single.sh
    Deleted:    cluster_scripts/bash_scripts/std_merfish_bc.sh
    Deleted:    cluster_scripts/bash_scripts/std_merfish_liver.sh
    Deleted:    cluster_scripts/bash_scripts/std_merfish_preoptic.sh
    Deleted:    cluster_scripts/bash_scripts/std_mibi_tnbc.sh
    Deleted:    cluster_scripts/bash_scripts/std_synthetic.sh
    Deleted:    cluster_scripts/bash_scripts/tboost_mibi_tnbc.sh
    Deleted:    cluster_scripts/bash_scripts/two_args_caller.sh
    Modified:   cluster_scripts/benchmark/four_i.R
    Deleted:    cluster_scripts/benchmark/hvg_merfish_bc.R
    Deleted:    cluster_scripts/benchmark/hvg_merfish_liver.R
    Deleted:    cluster_scripts/benchmark/hvg_merfish_preoptic.R
    Deleted:    cluster_scripts/benchmark/std_merfish_bc.R
    Deleted:    cluster_scripts/benchmark/std_merfish_liver.R
    Deleted:    cluster_scripts/benchmark/std_merfish_preoptic.R
    Deleted:    cluster_scripts/benchmark/std_mibi_tnbc.R
    Modified:   cluster_scripts/benchmark/synthetic.R
    Modified:   cluster_scripts/preprocessing/four_i.prep.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/introduction.Rmd</code>) and HTML (<code>docs/introduction.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/e9c007647ea473f2118b7e74517932deb1b93556/docs/introduction.html" target="_blank">e9c0076</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/f1596f7494ec85401df7a70387423990a59d27fe/docs/introduction.html" target="_blank">f1596f7</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/01d492549edefba18a95ca78b06a71d4025fcda3/analysis/introduction.Rmd" target="_blank">01d4925</a>
</td>
<td>
schae211
</td>
<td>
2021-12-18
</td>
<td>
wflow_publish(“analysis/introduction.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/52b8d6094a2ad9e9ff6f71de3839000a6315f317/docs/introduction.html" target="_blank">52b8d60</a>
</td>
<td>
schae211
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/d104aae3e8ff93387a0e1c2ebdc74f9265fdc1ce/analysis/introduction.Rmd" target="_blank">d104aae</a>
</td>
<td>
schae211
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(“analysis/introduction.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/1710d804182e867e8d3dc0d236fbf2f976201c9e/docs/introduction.html" target="_blank">1710d80</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/eb7b59d4c30a956379c497e2051bae1422291954/analysis/introduction.Rmd" target="_blank">eb7b59d</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/b5b2de7a9fc8463be559d515e2cc10ff94dd631e/analysis/introduction.Rmd" target="_blank">b5b2de7</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
added license and new cluster files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/schae211/report/f60ac7ce61c41369d4d11e42b6ed80c2406dd63d/docs/introduction.html" target="_blank">f60ac7c</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/schae211/report/blob/e11220706b944e0a6e2f52fa7f198731c5ef237b/analysis/introduction.Rmd" target="_blank">e112207</a>
</td>
<td>
schae211
</td>
<td>
2021-12-15
</td>
<td>
wflow_publish(“analysis/introduction.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify;
margin: 300px}
</style>
<p>In recent years the field of single-cell biology has vastly expanded. Since the publication of Drop-seq<span class="citation">(<a href="#ref-10.1016/j.cell.2015.05.002" role="doc-biblioref"><em>1</em></a>)</span> in 2015, the number of studies based on single-cell omics has exploded, primarily due to the commercialization of microfluidics-based single-cell sequencing by 10X Genomics. However, this technology requires a single-cell suspension, meaning tissues must be dissociated. Consequently, we loose information about the spatial context of cells, which is crucial for their function. Cells use different mechanisms to exchange information such as juxtacrine signaling via membrane-bound ligands or paracrine signaling via secreted ligands. Thereby, most communication mechanisms are limited to the neighborhood of a given cell due to dilution of signals by diffusion or the requirement of direct cell-to-cell contact. Therefore, inferring cell-cell communcation (CCC) or the interaction of different markers using single-cell expression data based on dissociated tissues is difficult.</p>
<p>Taking the spatial context into consideration should help to infer interactions. This requires multiplexed spatially resolved data and adequate analysis frameworks. Imaging single molecules in single cells has been a long-standing interest in biology, but early approaches were limited to a few markers. Only in the last years, there has been a surge of spatially resolved transcriptomic and proteomic approaches, which I will briefly describe below based on these reviews <span class="citation">(<a href="#ref-Moses2021.05.11.443152" role="doc-biblioref"><em>2</em></a>, <a href="#ref-10.3389/fimmu.2019.02657" role="doc-biblioref"><em>3</em></a>)</span>. (should I briefly describe MISTy here?)</p>
<div id="spatial-transcriptomics" class="section level1">
<h1>1.1 Spatial Transcriptomics</h1>
<p>Modern spatial transcriptomic approaches can be broadly classified into four categories, namely microdissection, single-molecule fluorescent in-situ hybridization (smFISH), in-situ sequencing (ISS), and spatial barcoding in conjunction with next-generation sequencing (NGS). Microdissection refers to a couple of methods for isolating small regions of interest which can subsequently be analyzed (for example by NGS). Advantages are the flexibility with respect to the size of the isolated spots (e.g. down to the size of nuclei) and downstream analysis, and the compatibility with FFPE tissues. However, this comes at the cost of being laborious due to sequential handling of the isolated spots. Thus, the throughput, meaning how many cells/spots can be processed in a given time, is much higher for the other methods mentioned above. In particular spatial barcoding with NGS offers a high throughput due to its commercialization by 10X Genomics (“Visium”) and standard downstream pipelines of NGS data (demultiplexing, alignment, etc.).</p>
<p>Visium is based on slides containing spots coated with capture sequences comprising a sequencing handle, barcode, UMI, and poly-T. The tissue is added on top of the slide and permeabilized, such that the polyadenylated transcripts bind to the capture sequences. Being based on NGS, these methods cover the whole transcriptome, which comes at the expense of a low resolution (corresponding to the size of the spots).</p>
<p>Both smFISH and ISS have single-cell (or even subcellular) resolution. Two commong methods for smFISH are seqFISH <span class="citation">(<a href="#ref-10.1016/j.neuron.2016.10.001" role="doc-biblioref"><em>4</em></a>)</span> and MERFISH <span class="citation">(<a href="#ref-10.1073/pnas.1612826113" role="doc-biblioref"><em>5</em></a>)</span>. They are both based on multiple rounds of hybridization. seqFISH uses unique sequences of colors in each hybridization round to identify transcripts (i.e. in each round a transcript is labeled with multiple probes containing the same fluorophore). MERFISH uses transcript specific encoding probes which can be bound by fluorescently labeled readout probes which always have the some color. By construction, a readout probe binds or does not bind a given transcript in a given hybridization round, creating a binary code of the same length as the number of hybridzation rounds. So each transcript is encoded by a unique binary code. By choosing a Hamming distance larger than one, this approach helps to reduce detection errors. Usually both methods cover around 300 different transcripts, but can be scaled up to cover many more genes (usually at the cost of including fewer cells).</p>
<p>The last category of spatial transcriptomic technologies discussed here comprises ISS, which can be divided into sequencing by ligation (SBL) and sequencing by synthesis (SBS). Most ISS technolgies, such as the Cartana platform (10X Genomics) are based on SBL. The specificity of DNA ligation is exploited by padlock probes which can only be ligated if they perfectly bind to reverse transcribed RNAs. Upon ligation the padlock probes are amplified by rolling-circle amplification (RCA) and detected by fluorescence labeling. <span class="citation">(<a href="#ref-10.1016/j.cell.2020.06.038" role="doc-biblioref"><em>6</em></a>)</span> Due to RCA signal amplification the imaging can be done at lower magnifications compared to smFISH methods allowing for larger tissue sections to be imaged.</p>
<div id="spatial-proteomics" class="section level2">
<h2>1.2 Spatial Proteomics</h2>
<p>Apart from spatial transcriptomics, the field of spatial proteomics is also rapidly expanding. Two examples for spatial proteomic technologies are imaging mass cytometry (IMC) <span class="citation">(<a href="#ref-10.1038/nmeth.2869" role="doc-biblioref"><em>7</em></a>)</span> and multiplexed ion beam imaging (MIBI) <span class="citation">(<a href="#ref-10.1038/nm.3488" role="doc-biblioref"><em>8</em></a>)</span>. Conventional technologies to visualize proteins in tissue sections such as immunohistochemistry (IHC) are based on antibodies conjugated with fluorophores or reporter enzymes, which makes mulitplexing difficult due to spectral overlap. For IHC several rounds of staining, imaging and photobleaching are needed for multiplexing, whereas IMC and MIBI solve this issue by using antibodies conjugated with heavy-metal isotopes which can be detected via mass cytometry imaging. Both technologies can image up to 40 different proteins. MIBI uses an oxygen duoplasmatron ion beam to ionize the heavy-metal labels, whereas IMC uses an UV laser to abalte the tissue which is subsequently atomized and ionized. According to Baharlou H. et al <span class="citation">(<a href="#ref-10.3389/fimmu.2019.02657" role="doc-biblioref"><em>3</em></a>)</span>, the main differences between the technologies is that the size of the MIBI ion beam and thus the spatial resolution is tunable between 1,000 and 260 nm, whereas the IMC laser is fixed at 1,000 nm. However, IMC has faster acquisition times compared to MIBI.</p>
</div>
<div id="data-analysis" class="section level2">
<h2>1.3 Data Analysis</h2>
<p>No matter which of the above technologies is used, a table containing either the markers for each spatial unit (spot or single cells) or the coordinates of markers (high resolution but no segmentation) is obtained after preprocessing. Following normalilzation, standard scRNA-seq tools are often used to explore the data. However, as more spatially resolved datasets are being published, new tools are developed to exploit richer information. For example the detection of spatially variable genes (e.g. SPARK-X <span class="citation">(<a href="#ref-10.1186/s13059-021-02404-0" role="doc-biblioref"><em>9</em></a>)</span>, SpatialDE <span class="citation">(<a href="#ref-10.1038/nmeth.4636" role="doc-biblioref"><em>10</em></a>)</span>), the detection of co-expression patterns/gene programs (e.g. ), and the detection of cellular niches (e.g. ).</p>
<p>Upon identification of interesting patterns in spatially resolved omics data one might as the question what drives the emergence or stabilizes these patterns? Is the spatial structure driven by self-organization, juxtacrine/ paracrine signaling, or broad gradients of morphogens? Several tools have been published which tackle related questions. Spatial Variance Component Analysis (SVCA) <span class="citation">(<a href="#ref-10.1016/j.celrep.2019.08.077" role="doc-biblioref"><em>11</em></a>)</span> for example decomposed the variance of a given marker into an intrinsic, intercellular, and environmental component. However, these spatial contexts are fixed and no information about which other markers can explain the expression of a given marker are learned.</p>
<p>The Multiview Intercellular SpaTial modeling framework (MISTy) <span class="citation">(<a href="#ref-Tanevski2020.05.08.084145" role="doc-biblioref"><em>12</em></a>)</span> on the other hand is much more flexible, allowing users to define the spatial context however they want. Additionally MISTy is build around an explainable machine learning algorithm faciliatating interpretability of the results. More specifically</p>
<p>Show case here the primary figure we get from MISTy</p>
<p>Use my presentation here :)</p>
<p>More importantly, one can ask questions, such as how diseases and tissue disorganization are associated.</p>
<ul>
<li><p>Include the theis model <span class="citation">(<a href="#ref-Fischer2021.07.11.451750" role="doc-biblioref"><em>13</em></a>)</span></p></li>
<li><p>MISTy is an explainable machine learning framework to probe intra- and intercellular marker interactions.</p></li>
<li><p>Assuming we have different markers (proteins, transcripts, etc.).</p></li>
</ul>
<p><strong>1. Question</strong>: Can we predict the expression of marker X using the information about the expression of other markers within the same cell (juxtaview)</p>
<p><strong>2. Question</strong>: Does our prediction improve if we take the expression of the other markers in the spatial context into consideration?</p>
<ul>
<li>Thereby we can define spatial context flexibly, meaning only the direct neighbors (juxtaview) or the broader tissue (paraview), (or actually any other view you could think of).</li>
</ul>
<p><strong>3. Question</strong>: What kind of spatial context is important to improve our prediction?</p>
<p><strong>4. Question</strong>: Which other markers are helpful to predict the expression of marker X?</p>
<ul>
<li>If another marker Y is important to predict the expression of markers, there is some sort of relationship between the markers (which has neither to be causal nor linear!).</li>
</ul>
<p><img src="https://www.dropbox.com/s/5wyh520i1wl62ul/graphical_abstract.png?raw=1" align="center" width="500"></p>
</div>
<div id="this-work" class="section level2">
<h2>1.4 This Work</h2>
<p>In the <a href="https://schae211.github.io/report/tutorial.html">Methods</a> section I will describe the flexible implementation of different machine learning algorithms for training view-specific models in MISTy. Based on the new underlaying framework, the function to model each view is now passed as an argument to the <code>run_misty()</code> call. The default model is still random forest, but now users can either plug in another provided model function such as gradient boosting or define their own function such as non-negative least-squares.</p>
<pre class="r"><code>misty.views %&gt;%
  run_misty(model.function = gradient_boosting_model, 
            booster = &quot;gbtree&quot;)</code></pre>
<details>
<summary>
Click here for MISTy flowchart
</summary>
<div id="htmlwidget-06bbab40e228689d42ad" style="width:672px;height:480px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-06bbab40e228689d42ad">{"x":{"diagram":"digraph{\n\n      # setting the global strucutre (top to low)\n      graph[rankdir = TL]\n  \n      # setting the default node layout\n      node[shape = rectangle, style = filled]\n  \n      subgraph cluster_1 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"Input: Spatial Omics\"\n        node[fillcolor = White, margin = 0.2]\n          EXP[label = \"expression\"]\n          POS[label = \"positions\"]\n      }\n      \n      subgraph cluster_3 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"Generating Views\"\n        node[fillcolor = White, margin = 0.2]\n        CIV[label = \"create_initial_view()\"]\n        AJV[label = \"add_juxtaview()\"]\n        APV[label = \"add_paraview()\"]\n      }\n      \n      subgraph cluster_4 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"MISTy Views\"\n        node[fillcolor = White, margin = 0.2]\n        IV[label = \"intraview\"]\n        JV[label = \"juxtaview\"]\n        PV[label = \"paraview\"]\n        V[label = \"misty views\"]\n      }\n      \n      subgraph cluster_5 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"MISTy Parameters\"\n        node[fillcolor = White, margin = 0.2]\n        RES[label = \"results.folder\"]\n        MF[label = \"model.function\", fillcolor = \"Gold\"]\n        TS[label = \"target.subset\"]\n        CV[label = \"cv.folds\"]\n        BI[label = \"bypass.intra\"]\n        DOTS[label = \"...\"]\n      }\n      \n      subgraph cluster_6 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"Output: MISTy Results\"\n        node[fillcolor = White, margin = 0.2]\n          PER[label = \"performance\"]\n          CON[label = \"contribution\"]\n          IMP[label = \"importances\"]\n      }\n      \n      subgraph cluster_8 {\n        graph[shape = rectangle]\n        style = rounded\n        bgcolor = PaleTurquoise\n        label = \"Running MISTy\"\n        node[fillcolor = White, margin = 0.2]\n        RM[label = \"run_misty()\"]\n        CR[label = \"collect_results()\"]\n      }\n      \n      edge[color = black, arrowhead = vee, arrowsize = 0.8]\n      EXP -> CIV\n      POS -> {APV, AJV}\n      CIV -> {APV, AJV}\n      CIV -> IV\n      APV -> PV\n      AJV -> JV\n      {IV, PV, JV} -> V\n      {V, RES, MF, TS, CV, BI, DOTS} -> RM\n      RM -> CR\n      CR -> {PER, CON, IMP}\n      \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
</details>
<p>In the results section, I will first show <a href="https://schae211.github.io/report/benchmark.html">benchmarking</a> results, where I compare the performance of different algorithms on different spatial omics datasets. Furthermore, I explore the impact of different hyperparameters on the model performance for the algorithms.</p>
<p>In the <a href="https://schae211.github.io/report/frameworks.html">frameworks</a> section I briefly describe we could not use one of the major machine learning frameworks in R, namely <code>caret</code>, <code>tidymodels</code>, and <code>mlr3</code> as backend in MISTy models. These frameworks are either not flexible enough, or their performance is bad.</p>
<p>Also, I will describe an <a href="https://schae211.github.io/report/model_api.html">alternative</a> implementation of a new MISTy framework that allows users to select more ML models than only random forests, but not to use functions defined by themselves. Running MISTy with an ensemble of 50 MARS models (interaction term 2) trained with bootstrap samples would looks like this.</p>
<pre class="r"><code>misty.views %&gt;% 
  run_misty(method = &quot;bag&quot;, learner = &quot;earth&quot;, 
            n.learners = 50, degree = 2)</code></pre>
<p>In the end, I will <a href="https://schae211.github.io/report/model_api.html">discuss</a> the results of my work and pose remaining open questions for the development of MISTy and the field of spatial omics technologies.</p>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body">
<div id="ref-10.1016/j.cell.2015.05.002" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">E. Z. Macosko, A. Basu, R. Satija, J. Nemesh, K. Shekhar, M. Goldman, I. Tirosh, A. R. Bialas, N. Kamitaki, E. M. Martersteck, J. J. Trombetta, D. A. Weitz, J. R. Sanes, A. K. Shalek, A. Regev, S. A. McCarroll, <span class="nocase">Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</span>. <em>Cell</em>. <strong>161</strong>, 1202–1214 (2015).</div>
</div>
<div id="ref-Moses2021.05.11.443152" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">L. Moses, L. Pachter, Museum of spatial transcriptomics. <em>bioRxiv</em> (2021), doi:<a href="https://doi.org/10.1101/2021.05.11.443152">10.1101/2021.05.11.443152</a>.</div>
</div>
<div id="ref-10.3389/fimmu.2019.02657" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">H. Baharlou, N. P. Canete, A. L. Cunningham, A. N. Harman, E. Patrick, <span class="nocase">Mass Cytometry Imaging for the Study of Human Diseases—Applications and Data Analysis Strategies</span>. <em>Frontiers in Immunology</em>. <strong>10</strong>, 2657 (2019).</div>
</div>
<div id="ref-10.1016/j.neuron.2016.10.001" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">S. Shah, E. Lubeck, W. Zhou, L. Cai, <span class="nocase">In Situ Transcription Profiling of Single Cells Reveals Spatial Organization of Cells in the Mouse Hippocampus</span>. <em>Neuron</em>. <strong>92</strong>, 342–357 (2016).</div>
</div>
<div id="ref-10.1073/pnas.1612826113" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">J. R. Moffitt, J. Hao, G. Wang, K. H. Chen, H. P. Babcock, X. Zhuang, <span class="nocase">High-throughput single-cell gene-expression profiling with multiplexed error-robust fluorescence in situ hybridization</span>. <em>Proceedings of the National Academy of Sciences</em>. <strong>113</strong>, 11046–11051 (2016).</div>
</div>
<div id="ref-10.1016/j.cell.2020.06.038" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">W.-T. Chen, A. Lu, K. Craessaerts, B. Pavie, C. S. Frigerio, N. Corthout, X. Qian, J. Laláková, M. Kühnemund, I. Voytyuk, L. Wolfs, R. Mancuso, E. Salta, S. Balusu, A. Snellinx, S. Munck, A. Jurek, J. F. Navarro, T. C. Saido, I. Huitinga, J. Lundeberg, M. Fiers, B. D. Strooper, <span class="nocase">Spatial Transcriptomics and In Situ Sequencing to Study Alzheimer’s Disease</span>. <em>Cell</em>. <strong>182</strong>, 976–991.e19 (2020).</div>
</div>
<div id="ref-10.1038/nmeth.2869" class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">C. Giesen, H. A. O. Wang, D. Schapiro, N. Zivanovic, A. Jacobs, B. Hattendorf, P. J. Schüffler, D. Grolimund, J. M. Buhmann, S. Brandt, Z. Varga, P. J. Wild, D. Günther, B. Bodenmiller, <span class="nocase">Highly multiplexed imaging of tumor tissues with subcellular resolution by mass cytometry</span>. <em>Nature Methods</em>. <strong>11</strong>, 417–422 (2014).</div>
</div>
<div id="ref-10.1038/nm.3488" class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">M. Angelo, S. C. Bendall, R. Finck, M. B. Hale, C. Hitzman, A. D. Borowsky, R. M. Levenson, J. B. Lowe, S. D. Liu, S. Zhao, Y. Natkunam, G. P. Nolan, <span class="nocase">Multiplexed ion beam imaging of human breast tumors</span>. <em>Nature Medicine</em>. <strong>20</strong>, 436–442 (2014).</div>
</div>
<div id="ref-10.1186/s13059-021-02404-0" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">J. Zhu, S. Sun, X. Zhou, <span class="nocase">SPARK-X: non-parametric modeling enables scalable and robust detection of spatial expression patterns for large spatial transcriptomic studies</span>. <em>Genome Biology</em>. <strong>22</strong>, 184 (2021).</div>
</div>
<div id="ref-10.1038/nmeth.4636" class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">V. Svensson, S. A. Teichmann, O. Stegle, <span class="nocase">SpatialDE: identification of spatially variable genes</span>. <em>Nature methods</em>. <strong>15</strong>, 343–346 (2018).</div>
</div>
<div id="ref-10.1016/j.celrep.2019.08.077" class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">D. Arnol, D. Schapiro, B. Bodenmiller, J. Saez-Rodriguez, O. Stegle, <span class="nocase">Modeling Cell-Cell Interactions from Spatial Molecular Data with Spatial Variance Component Analysis</span>. <em>Cell Reports</em>. <strong>29</strong>, 202–211.e6 (2019).</div>
</div>
<div id="ref-Tanevski2020.05.08.084145" class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">J. Tanevski, R. O. R. Flores, A. Gabor, D. Schapiro, J. Saez-Rodriguez, Explainable multi-view framework for dissecting intercellular signaling from highly multiplexed spatial data. <em>bioRxiv</em> (2021), doi:<a href="https://doi.org/10.1101/2020.05.08.084145">10.1101/2020.05.08.084145</a>.</div>
</div>
<div id="ref-Fischer2021.07.11.451750" class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">D. S. Fischer, A. C. Schaar, F. J. Theis, Learning cell communication from spatial graphs of cells. <em>bioRxiv</em> (2021), doi:<a href="https://doi.org/10.1101/2021.07.11.451750">10.1101/2021.07.11.451750</a>.</div>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7         pillar_1.6.3       compiler_4.1.2     bslib_0.3.1       
 [5] later_1.3.0        RColorBrewer_1.1-2 jquerylib_0.1.4    git2r_0.28.0      
 [9] workflowr_1.6.2    tools_4.1.2        digest_0.6.28      jsonlite_1.7.2    
[13] evaluate_0.14      lifecycle_1.0.1    tibble_3.1.5       pkgconfig_2.0.3   
[17] rlang_0.4.12       rstudioapi_0.13    yaml_2.2.1         xfun_0.26         
[21] fastmap_1.1.0      DiagrammeR_1.0.6.1 stringr_1.4.0      knitr_1.36        
[25] fs_1.5.0           vctrs_0.3.8        sass_0.4.0         htmlwidgets_1.5.4 
[29] rprojroot_2.0.2    glue_1.4.2         R6_2.5.1           fansi_0.5.0       
[33] rmarkdown_2.11     magrittr_2.0.1     whisker_0.4        promises_1.2.0.1  
[37] ellipsis_0.3.2     htmltools_0.5.2    httpuv_1.6.3       utf8_1.2.2        
[41] stringi_1.7.5      visNetwork_2.1.0   crayon_1.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
