---
title: "Introduction"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: no
    toc_float: no
bibliography: references.bib
csl: science.csl
---

This report documents what I have worked on between September 2021
and December 2021 in the [Saez-Rodriguez Group](https://saezlab.org) supervised
by Jovan Tanevski.

See the corresponding
[GitHub repository](https://github.com/schae211/mistyR/tree/as-function).

# Introduction

In recent years the field of single-cell biology has vastly expanded. Since the publication of Drop-seq[10.1016/j.cell.2015.05.002] in 2015, the number of studies based on single-cell omics has exploded, primarily due to the  commercialization of microfluidics-based single-cell sequencing by 10X Genomics. However, to use this technology a single-cell suspension must be used, meaning tissues must for example be dissociated before sequencing. Consequently, we loose information about the spatial context, which is crucial for cellular function. Different mechanisms are employed by cells to exchange information such as juxtacrine signaling via membrane-bound ligands or paracrine signaling via secreted ligands. Apart from hormonal signaling, most communication mechanisms are limited to the neighborhood of a given cell due to dilution of signals by diffusion or the requirement of direct cell-to-cell contact. Therefore, infering cell-cell-communcation (CCC) using single-cell expression data based on dissociated tissues is difficult. 

Conversely, taking the spatial context into consideration should help to infer interactions. This requires multiplexed spatially resolved data, which only became available in the last years. Early approaches to investigate the expression of single cells were limited to a few markers, but 

- Spatial Omics

-- Describing the 3 kind of spatial transcriptomic technologes we have
[@Moses2021.05.11.443152]
-- Describing the kind of proteomic technologies we have
(describing the features of the different technolgies)

- Mention SVCA for example (as another method). (I could orient the
introduction somewhat on the MISTy paper)

- What is MISTy
[@Tanevski2020.05.08.084145]

# Tutorial

This Rmarkdown showcases the new implementation of MISTy, where the
function to model each view are passed as an argument to the `run_misty()` 
call. 

```{r eval=FALSE}
misty.views %>% misty.views %>%
  run_misty(model.function = bagged_mars_model, n.bags = 50, degree = 2)
```

The tutorial includes also a section on how to construct those function
(i.e. what is the input and required output).

# Benchmark

This Rmarkdown contains:

a) The results of comparing the performance of different ML algorithms on 
different spatial omics datasets.

b) The results of testing different hyperparameters for certain algorithms
on different spatial omics datasets.

# Frameworks

This Rmarkdown describes why I choose not to use any existing ML framework in
MISTy. In a nutshell, these frameworks are either too slow or too unflexible 
(e.g. not able to return out-of-bag estimates).

# Model API

This Rmarkdown describes another way of generalizing the training of 
view-specific models, such that the call looks like this:

```{r eval=FALSE}
misty.views %>% run_misty(method = "bag", learner = "earth", 
                          n.learners = 50, degree = 2)
```

# Conclusions

This Rmarkdown discusses the results of my internship and poses open 
questions for the MISTy and the field of spatial omics technologies. 

# Random

This Rmarkdown contains results I could not fit into the other categories
thus far.
